<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.47">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Dylan Padilla">

<title>index</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="index_files/libs/clipboard/clipboard.min.js"></script>
<script src="index_files/libs/quarto-html/quarto.js"></script>
<script src="index_files/libs/quarto-html/popper.min.js"></script>
<script src="index_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="index_files/libs/quarto-html/anchor.min.js"></script>
<link href="index_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="index_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="index_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="index_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="index_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
</div>


<div class="quarto-title-meta-author">
  <div class="quarto-title-meta-heading">Author</div>
  <div class="quarto-title-meta-heading">Affiliation</div>
  
    <div class="quarto-title-meta-contents">
    <p class="author">Dylan Padilla </p>
  </div>
  <div class="quarto-title-meta-contents">
        <p class="affiliation">
            Yale Institute for Biospheric Studies, Yale University, New Haven, CT 06511, USA.
          </p>
      </div>
  </div>

<div class="quarto-title-meta">

      
  
    
  </div>
  


</header>


<p>Full workflow of data analyses for:</p>
<p><strong>Reconstructing the Demographic History of a Freeze-Tolerant Frog: The Interplay Between Selection and <span class="math inline">\(N_e\)</span></strong></p>
<p><img src="imgs/yale-logo.png" class="img-fluid"></p>
<p><em>For a workflow of the genome annotation, please visit <a href="https://github.com/jodymtaft/Genome-Processing">https://github.com/jodymtaft/Genome-Processing</a></em></p>
<p><strong>Population genetic analyses across the range of the wood frog:</strong></p>
<p><strong>Ipyrad assembly pipeline</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co">#| output: true</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> ./scripts/params-aRanSyl.txt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>------- ipyrad params file (v.0.9.107)------------------------------------------
aRanSyl                        ## [0] [assembly_name]: Assembly name. Used to name output directories for assembly steps
/gpfs/gibbs/project/skelly/dp996/Wood-frog_ddRADseq/ipyrad-reference/      ## [1] [project_dir]: Project dir (made in curdir if not present)
                               ## [2] [raw_fastq_path]: Location of raw non-demultiplexed fastq files
                               ## [3] [barcodes_path]: Location of barcodes file
/gpfs/gibbs/project/skelly/dp996/Wood-frog_ddRADseq/fastq/*fastq.gz           ## [4] [sorted_fastq_path]: Location of demultiplexed/sorted fastq files
reference                      ## [5] [assembly_method]: Assembly method (denovo, reference)
/gpfs/gibbs/project/skelly/dp996/Wood-frog_ddRADseq/resources/GCA_028564925.1_aRanSyl1.merge_genomic.fna     ## [6] [reference_sequence]: Location of reference sequence file
ddrad                          ## [7] [datatype]: Datatype (see docs): rad, gbs, ddrad, etc.
GATC, ATGCA                    ## [8] [restriction_overhang]: Restriction overhang (cut1,) or (cut1, cut2)
5                              ## [9] [max_low_qual_bases]: Max low quality base calls (Q&lt;20) in a read
33                             ## [10] [phred_Qscore_offset]: phred Q score offset (33 is default and very standard)
6                              ## [11] [mindepth_statistical]: Min depth for statistical base calling
6                              ## [12] [mindepth_majrule]: Min depth for majority-rule base calling
10000                          ## [13] [maxdepth]: Max cluster depth within samples
0.85                           ## [14] [clust_threshold]: Clustering threshold for de novo assembly
0                              ## [15] [max_barcode_mismatch]: Max number of allowable mismatches in barcodes
1                              ## [16] [filter_adapters]: Filter for adapters/primers (1 or 2=stricter)
35                             ## [17] [filter_min_trim_len]: Min length of reads after adapter trim
2                              ## [18] [max_alleles_consens]: Max alleles per site in consensus sequences
0.05                           ## [19] [max_Ns_consens]: Max N's (uncalled bases) in consensus
0.05                           ## [20] [max_Hs_consens]: Max Hs (heterozygotes) in consensus
4                              ## [21] [min_samples_locus]: Min # samples per locus for output
0.2                            ## [22] [max_SNPs_locus]: Max # SNPs per locus
5                              ## [23] [max_Indels_locus]: Max # of indels per locus
0.5                            ## [24] [max_shared_Hs_locus]: Max # heterozygous sites per locus
0, 0, 0, 0                     ## [25] [trim_reads]: Trim raw read edges (R1&gt;, &lt;R1, R2&gt;, &lt;R2) (see docs)
0, 0, 0, 0                     ## [26] [trim_loci]: Trim locus edges (see docs) (R1&gt;, &lt;R1, R2&gt;, &lt;R2)
p, s, l, g, v, a                        ## [27] [output_formats]: Output formats (see docs)
                               ## [28] [pop_assign_file]: Path to population assignment file
                               ## [29] [reference_as_filter]: Reads mapped to this reference are removed in step 3</code></pre>
</div>
</div>
<p><strong>R libraries</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="do">## loading packages</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(adegenet)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ade4)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(car)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(canadamaps)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dartRverse)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dartR.popgen)</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ecodist)</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(hierfstat)</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenomicRanges)</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GO.db)</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(geodata)</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(LEA)</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(lfmm)</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(maps)</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mapplots)</span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mapproj)</span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(optparse)</span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Dr.eg.db)</span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Mm.eg.db)</span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Xl.eg.db)</span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pegas)</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plotrix)</span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(poppr)</span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(prettymapr)</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(qqman)</span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(qvalue)</span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(raster)</span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rnaturalearth)</span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sf)</span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scales)</span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SeqArray)</span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SeqVarTools)</span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(shape)</span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SNPRelate)</span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(terra)</span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(topGO)</span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan)</span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(viridis)</span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vcfR)</span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(xtable)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>This script was designed to perform quality control of the aRanSyl ddRADseq samples</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="do">## reading .vcf file and producing genind object</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>my_vcf <span class="ot">&lt;-</span> <span class="fu">read.vcfR</span>(<span class="st">"/gpfs/gibbs/project/skelly/dp996/Wood-frog_ddRADseq/ipyrad-reference/aRanSyl_outfiles/aRanSyl_copy_reheadered_snps.vcf.gz"</span>)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>aRanSyl_genind <span class="ot">&lt;-</span> <span class="fu">vcfR2genind</span>(my_vcf)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>aRanSyl_genind</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="do">## Access coordinates from the vcfR object's fix slot</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>aRanSyl_genind<span class="sc">@</span>other<span class="sc">$</span>chromosome <span class="ot">&lt;-</span> my_vcf<span class="sc">@</span>fix[ , <span class="dv">1</span>]</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>aRanSyl_genind<span class="sc">@</span>other<span class="sc">$</span>position <span class="ot">&lt;-</span> my_vcf<span class="sc">@</span>fix[ , <span class="dv">2</span>]</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>aRanSyl_genind</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="do">## removing samples that did not pass the adapter content</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a><span class="do">## check qc from the multiqc_report.html file</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>ind <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"s_272_S121_R1_001"</span>, <span class="st">"s_346_S186_R1_001"</span>)</span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>aRanSyl_genind <span class="ot">&lt;-</span> aRanSyl_genind[<span class="sc">!</span><span class="fu">indNames</span>(aRanSyl_genind) <span class="sc">%in%</span> ind, ]</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a><span class="do">## calculate the percentage of complete genotypes per loci</span></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a><span class="do">## in the aRanSyl SNP data set</span></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a><span class="do">## remove loci with &gt; 20% missing data</span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>aRanSyl_genind <span class="ot">&lt;-</span> <span class="fu">missingno</span>(aRanSyl_genind, <span class="at">type =</span> <span class="st">"loc"</span>, <span class="at">cutoff =</span> <span class="fl">0.20</span>)</span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a><span class="do">## calculate the percentage of complete genotypes per</span></span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a><span class="do">## individual in the aRanSyl SNP data set</span></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a><span class="do">## remove individuals with &gt; 20% missing genotypes</span></span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a>aRanSyl_genind <span class="ot">&lt;-</span> <span class="fu">missingno</span>(aRanSyl_genind, <span class="at">type =</span> <span class="st">"geno"</span>, <span class="at">cutoff =</span> <span class="fl">0.20</span>)</span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a><span class="do">## filtering out loci with &gt; 2 allelles</span></span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">which</span>(<span class="fu">nAll</span>(aRanSyl_genind) <span class="sc">==</span> <span class="dv">2</span>))</span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a>aRanSyl_genind <span class="ot">&lt;-</span> aRanSyl_genind[loc <span class="ot">=</span> n]</span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a>aRanSyl_genind</span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a><span class="do">## print the number of multilocus genotypes</span></span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true" tabindex="-1"></a><span class="fu">mlg</span>(aRanSyl_genind)</span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true" tabindex="-1"></a><span class="fu">isPoly</span>(aRanSyl_genind) <span class="sc">%&gt;%</span> summary</span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true" tabindex="-1"></a>poly_loci_aRanSyl <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">which</span>(<span class="fu">isPoly</span>(aRanSyl_genind) <span class="sc">==</span> <span class="cn">TRUE</span>))</span>
<span id="cb4-45"><a href="#cb4-45" aria-hidden="true" tabindex="-1"></a>aRanSyl_genind <span class="ot">&lt;-</span> aRanSyl_genind[loc <span class="ot">=</span> poly_loci_aRanSyl]</span>
<span id="cb4-46"><a href="#cb4-46" aria-hidden="true" tabindex="-1"></a><span class="fu">isPoly</span>(aRanSyl_genind) <span class="sc">%&gt;%</span> summary</span>
<span id="cb4-47"><a href="#cb4-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-48"><a href="#cb4-48" aria-hidden="true" tabindex="-1"></a><span class="do">## convert genind object to a data frame</span></span>
<span id="cb4-49"><a href="#cb4-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-50"><a href="#cb4-50" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(aRanSyl_genind, <span class="at">file =</span> <span class="st">"/gpfs/gibbs/project/skelly/dp996/Wood-frog_ddRADseq/data/aRanSyl_genind_chr_pos.RData"</span>)</span>
<span id="cb4-51"><a href="#cb4-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-52"><a href="#cb4-52" aria-hidden="true" tabindex="-1"></a><span class="do">## identify duplicated genotypes</span></span>
<span id="cb4-53"><a href="#cb4-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-54"><a href="#cb4-54" aria-hidden="true" tabindex="-1"></a>dups_aRanSyl <span class="ot">&lt;-</span> <span class="fu">mlg.id</span>(aRanSyl_genind)</span>
<span id="cb4-55"><a href="#cb4-55" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> dups_aRanSyl){ <span class="co"># for each element in the list object</span></span>
<span id="cb4-56"><a href="#cb4-56" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(dups_aRanSyl[i]) <span class="sc">&gt;</span> <span class="dv">1</span>){ <span class="co"># if the length is greater than 1</span></span>
<span id="cb4-57"><a href="#cb4-57" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(i) <span class="co"># print individuals that are duplicates</span></span>
<span id="cb4-58"><a href="#cb4-58" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb4-59"><a href="#cb4-59" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-60"><a href="#cb4-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-61"><a href="#cb4-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-62"><a href="#cb4-62" aria-hidden="true" tabindex="-1"></a><span class="do">## preparing .geno file for pop structure analysis</span></span>
<span id="cb4-63"><a href="#cb4-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-64"><a href="#cb4-64" aria-hidden="true" tabindex="-1"></a>aRanSyl_gl <span class="ot">&lt;-</span> <span class="fu">gi2gl</span>(aRanSyl_genind)</span>
<span id="cb4-65"><a href="#cb4-65" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(aRanSyl_gl, <span class="at">file =</span> <span class="st">"/gpfs/gibbs/project/skelly/dp996/Wood-frog_ddRADseq/data/aRanSyl_gl_chr_pos.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>This script was designed to compute pairwise Fst values across populations of aRanSyl</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"/gpfs/gibbs/project/skelly/dp996/Wood-frog_ddRADseq/data/aRanSyl_genind_pops.RData"</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"/gpfs/gibbs/project/skelly/dp996/Wood-frog_ddRADseq/data/aRanSyl_gl_pops.RData"</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Computing pairwise Fst test</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>aRanSyl_fst <span class="ot">&lt;-</span> <span class="fu">genet.dist</span>(aRanSyl_genind_pops, <span class="at">method =</span> <span class="st">"WC84"</span>) <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="do">## saving here just in case the commands below fail</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(aRanSyl_fst,</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>     <span class="at">file =</span> <span class="st">"/gpfs/gibbs/project/skelly/dp996/Wood-frog_ddRADseq/data/aRanSyl_fst.RData"</span>)</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>aRanSyl_fst_mat <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(aRanSyl_fst)</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>aRanSyl_fst_mat[aRanSyl_fst_mat <span class="sc">&lt;</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>aRanSyl_fst_mat[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>] <span class="do">## I ran this script using 1:10</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a><span class="do">## in this command and it gave me an error because there</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a><span class="do">## are only 9 columns, not 10</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(aRanSyl_fst_mat,</span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>     <span class="at">file =</span> <span class="st">"/gpfs/gibbs/project/skelly/dp996/Wood-frog_ddRADseq/data/aRanSyl_fst_mat.RData"</span>)</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>aRanSyl_pval_fst <span class="ot">&lt;-</span> <span class="fu">gl.fst.pop</span>(aRanSyl_gl_pops, <span class="at">nboots =</span> <span class="dv">1000</span>,</span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>                               <span class="at">percent =</span> <span class="dv">95</span>, <span class="at">nclusters =</span> <span class="dv">1</span>, <span class="at">verbose =</span> <span class="cn">NULL</span>)</span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a><span class="do">## saving here just in case the commands below fail</span></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(aRanSyl_pval_fst,</span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>     <span class="at">file =</span> <span class="st">"/gpfs/gibbs/project/skelly/dp996/Wood-frog_ddRADseq/data/aRanSyl_pval_fst.RData"</span>)</span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a>aRanSyl_fst <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">as.dist</span>(aRanSyl_pval_fst<span class="sc">$</span>Fsts))</span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a>aRanSyl_fst[aRanSyl_fst <span class="sc">&lt;</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="dv">0</span> <span class="do">## I ran this script using</span></span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a><span class="do">## aRanSyl_pval_fst here instead, and it gave me an error</span></span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a><span class="do">## because the matrix that should have been used is aRanSyl_fst</span></span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a>aRanSyl_fst</span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(aRanSyl_fst,</span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a>     <span class="at">file =</span> <span class="st">"/gpfs/gibbs/project/skelly/dp996/Wood-frog_ddRADseq/data/aRanSyl_pval_mat.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>This script was designed to examine population structure of the aRanSyl samples</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="do">## loading project</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co">#load("/Users/dpadil10/ASU Dropbox/Dylan Padilla/Yale/aRanSyl-PopGen/data/aRanSyl_genind.RData")</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"/Users/dpadil10/ASU Dropbox/Dylan Padilla/Yale/aRanSyl-PopGen/data/aRanSyl_genind_chr_pos.RData"</span>)</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"/Users/dpadil10/ASU Dropbox/Dylan Padilla/Yale/aRanSyl-PopGen/data/aRanSyl_gl_chr_pos.RData"</span>)</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="co">#load("/Users/dpadil10/ASU Dropbox/Dylan Padilla/Yale/aRanSyl-PopGen/data/aRanSyl_gl.RData")</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="do">## Converting genlight object to geno object</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a><span class="fu">gl2geno</span>(aRanSyl_gl, <span class="at">outfile =</span> <span class="st">"aRanSyl_geno"</span>,</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="at">outpath =</span> <span class="st">"/gpfs/gibbs/project/skelly/dp996/Wood-frog_ddRADseq/data/"</span>, <span class="at">verbose =</span> <span class="cn">NULL</span>)</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Run snmf algorithm</span></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>snmf1 <span class="ot">&lt;-</span> <span class="fu">snmf</span>(<span class="st">"/gpfs/gibbs/project/skelly/dp996/Wood-frog_ddRADseq/data/aRanSyl_geno.geno"</span>,</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>             <span class="at">K =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>, <span class="co"># number of K ancestral populations to run</span></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>             <span class="at">repetitions =</span> <span class="dv">50</span>, <span class="co"># 50 repetitions for each K</span></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>             <span class="at">entropy =</span> <span class="cn">TRUE</span>, <span class="co"># calculate cross-entropy</span></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>             <span class="at">project =</span> <span class="st">"new"</span>)</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a><span class="do">## plotting entropy to determine K</span></span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(snmf1, <span class="at">type =</span> <span class="st">"n"</span>, <span class="at">xaxt =</span> <span class="st">"n"</span>, <span class="at">yaxt =</span> <span class="st">"n"</span>, <span class="at">las =</span> <span class="dv">1</span>)</span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a><span class="fu">grid</span>()</span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">new =</span> <span class="cn">TRUE</span>)</span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(snmf1, <span class="at">col =</span> <span class="st">"black"</span>, <span class="at">cex =</span> <span class="fl">1.5</span>, <span class="at">pch =</span> <span class="dv">19</span>, <span class="at">las =</span> <span class="dv">1</span>)</span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a><span class="fu">Arrows</span>(<span class="at">x =</span> <span class="dv">9</span>, <span class="at">y =</span> <span class="fl">0.295</span>, <span class="at">x1 =</span> <span class="dv">9</span>, <span class="at">y1 =</span> <span class="fl">0.305</span>, <span class="at">col =</span> <span class="st">"black"</span>, <span class="at">arr.type =</span> <span class="st">"triangle"</span>,</span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a><span class="at">code =</span> <span class="dv">1</span>, <span class="at">lwd =</span> <span class="fl">1.5</span>, <span class="at">arr.length =</span> <span class="fl">0.2</span>)</span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a><span class="do">## extract the cross-entropy of all runs where K = 9</span></span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a>ce <span class="ot">&lt;-</span> <span class="fu">cross.entropy</span>(snmf1, <span class="at">K =</span> <span class="dv">9</span>)</span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a>ce</span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-38"><a href="#cb6-38" aria-hidden="true" tabindex="-1"></a><span class="do">## find the run with the lowest cross-entropy</span></span>
<span id="cb6-39"><a href="#cb6-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-40"><a href="#cb6-40" aria-hidden="true" tabindex="-1"></a>lowest.ce <span class="ot">&lt;-</span> <span class="fu">which.min</span>(ce)</span>
<span id="cb6-41"><a href="#cb6-41" aria-hidden="true" tabindex="-1"></a>lowest.ce</span>
<span id="cb6-42"><a href="#cb6-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-43"><a href="#cb6-43" aria-hidden="true" tabindex="-1"></a><span class="do">## extract Q-matrix for the best run</span></span>
<span id="cb6-44"><a href="#cb6-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-45"><a href="#cb6-45" aria-hidden="true" tabindex="-1"></a>qmatrix <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">Q</span>(snmf1, <span class="at">K =</span> <span class="dv">9</span>, <span class="at">run =</span> lowest.ce))</span>
<span id="cb6-46"><a href="#cb6-46" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(qmatrix)</span>
<span id="cb6-47"><a href="#cb6-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-48"><a href="#cb6-48" aria-hidden="true" tabindex="-1"></a><span class="do">## since I did not define populations a priori, each individual</span></span>
<span id="cb6-49"><a href="#cb6-49" aria-hidden="true" tabindex="-1"></a><span class="do">## sampled will be treated as a unique population</span></span>
<span id="cb6-50"><a href="#cb6-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-51"><a href="#cb6-51" aria-hidden="true" tabindex="-1"></a>pops <span class="ot">&lt;-</span> <span class="fu">row.names</span>(<span class="fu">data.frame</span>(aRanSyl_genind))</span>
<span id="cb6-52"><a href="#cb6-52" aria-hidden="true" tabindex="-1"></a>pops <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(pops)</span>
<span id="cb6-53"><a href="#cb6-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-54"><a href="#cb6-54" aria-hidden="true" tabindex="-1"></a>aRanSyl_genind<span class="sc">$</span>pop <span class="ot">&lt;-</span> pops</span>
<span id="cb6-55"><a href="#cb6-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-56"><a href="#cb6-56" aria-hidden="true" tabindex="-1"></a><span class="do">## plotting admixture proportions</span></span>
<span id="cb6-57"><a href="#cb6-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-58"><a href="#cb6-58" aria-hidden="true" tabindex="-1"></a>qmplot <span class="ot">&lt;-</span> <span class="fu">cbind</span>(qmatrix, pops)</span>
<span id="cb6-59"><a href="#cb6-59" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(qmplot)</span>
<span id="cb6-60"><a href="#cb6-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-61"><a href="#cb6-61" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> qmplot</span>
<span id="cb6-62"><a href="#cb6-62" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"/Users/dpadil10/ASU Dropbox/Dylan Padilla/Yale/aRanSyl-PopGen/data/aRanSyl_metadata.csv"</span>)</span>
<span id="cb6-63"><a href="#cb6-63" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(meta)</span>
<span id="cb6-64"><a href="#cb6-64" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>id <span class="ot">&lt;-</span> meta<span class="sc">$</span>id</span>
<span id="cb6-65"><a href="#cb6-65" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>pops <span class="ot">&lt;-</span> meta<span class="sc">$</span>pop</span>
<span id="cb6-66"><a href="#cb6-66" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> dat[<span class="fu">order</span>(dat<span class="sc">$</span>pops), ]</span>
<span id="cb6-67"><a href="#cb6-67" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(dat)</span>
<span id="cb6-68"><a href="#cb6-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-69"><a href="#cb6-69" aria-hidden="true" tabindex="-1"></a><span class="do">## plotting panel</span></span>
<span id="cb6-70"><a href="#cb6-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-71"><a href="#cb6-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-72"><a href="#cb6-72" aria-hidden="true" tabindex="-1"></a><span class="fu">layout</span>(<span class="fu">matrix</span>(<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>,</span>
<span id="cb6-73"><a href="#cb6-73" aria-hidden="true" tabindex="-1"></a>                <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>,</span>
<span id="cb6-74"><a href="#cb6-74" aria-hidden="true" tabindex="-1"></a>                <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>,</span>
<span id="cb6-75"><a href="#cb6-75" aria-hidden="true" tabindex="-1"></a>                <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>), <span class="at">ncol =</span> <span class="dv">6</span>, <span class="at">nrow =</span> <span class="dv">4</span>, <span class="at">byrow =</span> <span class="cn">TRUE</span>))</span>
<span id="cb6-76"><a href="#cb6-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-77"><a href="#cb6-77" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mgp =</span> <span class="fu">c</span>(<span class="fl">2.5</span>, <span class="dv">1</span>, <span class="dv">0</span>))</span>
<span id="cb6-78"><a href="#cb6-78" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(<span class="fu">t</span>(dat[<span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>]), <span class="at">col =</span> RColorBrewer<span class="sc">::</span><span class="fu">brewer.pal</span>(<span class="dv">9</span>,<span class="st">"Paired"</span>),</span>
<span id="cb6-79"><a href="#cb6-79" aria-hidden="true" tabindex="-1"></a><span class="at">border =</span> <span class="cn">NA</span>, <span class="at">space =</span> <span class="dv">0</span>, <span class="at">xlab =</span> <span class="st">""</span>, <span class="at">xaxt =</span> <span class="st">"n"</span>,  <span class="at">ylab =</span> <span class="st">"Admixture proportion"</span>,</span>
<span id="cb6-80"><a href="#cb6-80" aria-hidden="true" tabindex="-1"></a><span class="at">las =</span> <span class="dv">1</span>, <span class="at">cex.lab =</span> <span class="fl">1.5</span>, <span class="at">cex.axis =</span> <span class="dv">1</span>)</span>
<span id="cb6-81"><a href="#cb6-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-82"><a href="#cb6-82" aria-hidden="true" tabindex="-1"></a><span class="do">## adding population labels to the axis:</span></span>
<span id="cb6-83"><a href="#cb6-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-84"><a href="#cb6-84" aria-hidden="true" tabindex="-1"></a>medians <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb6-85"><a href="#cb6-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-86"><a href="#cb6-86" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(dat<span class="sc">$</span>pops)){</span>
<span id="cb6-87"><a href="#cb6-87" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb6-88"><a href="#cb6-88" aria-hidden="true" tabindex="-1"></a>    <span class="fu">axis</span>(<span class="dv">1</span>, <span class="at">at =</span> <span class="fu">median</span>(<span class="fu">which</span>(dat<span class="sc">$</span>pops <span class="sc">==</span> dat<span class="sc">$</span>pops[i])), <span class="at">labels =</span> <span class="st">""</span>)</span>
<span id="cb6-89"><a href="#cb6-89" aria-hidden="true" tabindex="-1"></a>    medians <span class="ot">&lt;-</span> <span class="fu">c</span>(medians, <span class="fu">median</span>(<span class="fu">which</span>(dat<span class="sc">$</span>pops <span class="sc">==</span> dat<span class="sc">$</span>pops[i])))</span>
<span id="cb6-90"><a href="#cb6-90" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-91"><a href="#cb6-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-92"><a href="#cb6-92" aria-hidden="true" tabindex="-1"></a>names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">paste</span>(<span class="fu">rep</span>(<span class="st">"Pop"</span>, <span class="dv">9</span>), <span class="fu">seq</span>(<span class="dv">1</span>, <span class="dv">9</span>, <span class="dv">1</span>), <span class="at">sep =</span> <span class="st">""</span>))</span>
<span id="cb6-93"><a href="#cb6-93" aria-hidden="true" tabindex="-1"></a>names</span>
<span id="cb6-94"><a href="#cb6-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-95"><a href="#cb6-95" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">tapply</span>(<span class="fu">t</span>(dat[<span class="dv">1</span>]), <span class="fu">t</span>(dat<span class="sc">$</span>pops), length)</span>
<span id="cb6-96"><a href="#cb6-96" aria-hidden="true" tabindex="-1"></a>obj</span>
<span id="cb6-97"><a href="#cb6-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-98"><a href="#cb6-98" aria-hidden="true" tabindex="-1"></a><span class="fu">segments</span>(<span class="at">x0 =</span> <span class="dv">0</span>, <span class="at">y0 =</span> <span class="fl">0.01</span>,</span>
<span id="cb6-99"><a href="#cb6-99" aria-hidden="true" tabindex="-1"></a>         <span class="at">x1 =</span> <span class="dv">0</span>, <span class="at">y1 =</span> <span class="dv">1</span>, <span class="at">lty =</span> <span class="dv">2</span>, <span class="at">lwd =</span> <span class="dv">2</span>)</span>
<span id="cb6-100"><a href="#cb6-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-101"><a href="#cb6-101" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(obj)){</span>
<span id="cb6-102"><a href="#cb6-102" aria-hidden="true" tabindex="-1"></a>    sum <span class="ot">&lt;-</span> <span class="fu">sum</span>(obj[<span class="dv">1</span><span class="sc">:</span>i])</span>
<span id="cb6-103"><a href="#cb6-103" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(sum)</span>
<span id="cb6-104"><a href="#cb6-104" aria-hidden="true" tabindex="-1"></a>    <span class="fu">segments</span>(<span class="at">x0 =</span> sum, <span class="at">y0 =</span> <span class="fl">0.01</span>,</span>
<span id="cb6-105"><a href="#cb6-105" aria-hidden="true" tabindex="-1"></a>             <span class="at">x1 =</span> sum, <span class="at">y1 =</span> <span class="dv">1</span>, <span class="at">lty =</span> <span class="dv">2</span>, <span class="at">lwd =</span> <span class="dv">2</span>)</span>
<span id="cb6-106"><a href="#cb6-106" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-107"><a href="#cb6-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-108"><a href="#cb6-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-109"><a href="#cb6-109" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(<span class="at">x =</span> <span class="fu">as.numeric</span>(<span class="fu">unique</span>(<span class="fu">as.character</span>(medians))), <span class="at">y =</span> <span class="fu">par</span>(<span class="st">"usr"</span>)[<span class="dv">3</span>] <span class="sc">-</span> <span class="fl">0.06</span>,</span>
<span id="cb6-110"><a href="#cb6-110" aria-hidden="true" tabindex="-1"></a>     <span class="at">labels =</span> names, <span class="at">xpd =</span> <span class="cn">NA</span>, <span class="at">srt =</span> <span class="dv">35</span>, <span class="at">cex =</span> <span class="fl">0.8</span>, <span class="at">adj =</span> <span class="dv">1</span>)</span>
<span id="cb6-111"><a href="#cb6-111" aria-hidden="true" tabindex="-1"></a><span class="fu">mtext</span>(<span class="st">"Populations"</span>, <span class="at">side =</span> <span class="dv">1</span>, <span class="at">line =</span> <span class="fl">2.5</span>)</span>
<span id="cb6-112"><a href="#cb6-112" aria-hidden="true" tabindex="-1"></a><span class="fu">mtext</span>(<span class="st">"A"</span>, <span class="at">side =</span> <span class="dv">2</span>, <span class="at">at =</span> <span class="fl">1.2</span>,  <span class="at">line =</span> <span class="fl">3.2</span>, <span class="at">las =</span> <span class="dv">1</span>)</span>
<span id="cb6-113"><a href="#cb6-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-114"><a href="#cb6-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-115"><a href="#cb6-115" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"/Users/dpadil10/ASU Dropbox/Dylan Padilla/Yale/aRanSyl-PopGen/data/aRanSyl_fst_mat.RData"</span>)</span>
<span id="cb6-116"><a href="#cb6-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-117"><a href="#cb6-117" aria-hidden="true" tabindex="-1"></a>fst <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(aRanSyl_fst_mat)</span>
<span id="cb6-118"><a href="#cb6-118" aria-hidden="true" tabindex="-1"></a>fst[fst <span class="sc">&lt;</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb6-119"><a href="#cb6-119" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(fst) <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">rep</span>(<span class="st">"Pop"</span>, <span class="dv">9</span>), <span class="fu">seq</span>(<span class="dv">1</span>, <span class="dv">9</span>, <span class="dv">1</span>), <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb6-120"><a href="#cb6-120" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(fst) <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">rep</span>(<span class="st">"Pop"</span>, <span class="dv">9</span>), <span class="fu">seq</span>(<span class="dv">1</span>, <span class="dv">9</span>, <span class="dv">1</span>), <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb6-121"><a href="#cb6-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-122"><a href="#cb6-122" aria-hidden="true" tabindex="-1"></a>get_lower_tri<span class="ot">&lt;-</span><span class="cf">function</span>(cor_matrix){</span>
<span id="cb6-123"><a href="#cb6-123" aria-hidden="true" tabindex="-1"></a>  cor_matrix[<span class="fu">lower.tri</span>(cor_matrix, <span class="at">diag =</span> <span class="cn">TRUE</span>)]</span>
<span id="cb6-124"><a href="#cb6-124" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(cor_matrix)</span>
<span id="cb6-125"><a href="#cb6-125" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-126"><a href="#cb6-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-127"><a href="#cb6-127" aria-hidden="true" tabindex="-1"></a>lower_tri_matrix <span class="ot">&lt;-</span> fst[<span class="fu">lower.tri</span>(<span class="fu">round</span>(fst, <span class="dv">2</span>), <span class="at">diag =</span> <span class="cn">TRUE</span>)]</span>
<span id="cb6-128"><a href="#cb6-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-129"><a href="#cb6-129" aria-hidden="true" tabindex="-1"></a>my_palette <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">"lightblue"</span>, <span class="st">"red"</span>))(<span class="at">n =</span> <span class="dv">100</span>)</span>
<span id="cb6-130"><a href="#cb6-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-131"><a href="#cb6-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-132"><a href="#cb6-132" aria-hidden="true" tabindex="-1"></a><span class="do">## Create matrix</span></span>
<span id="cb6-133"><a href="#cb6-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-134"><a href="#cb6-134" aria-hidden="true" tabindex="-1"></a>fst_matrix <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="at">nrow=</span><span class="dv">9</span>, <span class="at">ncol=</span><span class="dv">9</span>)</span>
<span id="cb6-135"><a href="#cb6-135" aria-hidden="true" tabindex="-1"></a>fst_matrix[<span class="fu">lower.tri</span>(fst_matrix, <span class="at">diag=</span><span class="cn">TRUE</span>)] <span class="ot">&lt;-</span> <span class="fu">round</span>(lower_tri_matrix, <span class="dv">2</span>)</span>
<span id="cb6-136"><a href="#cb6-136" aria-hidden="true" tabindex="-1"></a>labs <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Pop"</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>)</span>
<span id="cb6-137"><a href="#cb6-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-138"><a href="#cb6-138" aria-hidden="true" tabindex="-1"></a><span class="do">## 2. Prepare the Plotting Area</span></span>
<span id="cb6-139"><a href="#cb6-139" aria-hidden="true" tabindex="-1"></a><span class="do">## This is the key fix: par(mar=...) sets the margins manually.</span></span>
<span id="cb6-140"><a href="#cb6-140" aria-hidden="true" tabindex="-1"></a><span class="do">## Order: Bottom, Left, Top, Right. </span></span>
<span id="cb6-141"><a href="#cb6-141" aria-hidden="true" tabindex="-1"></a><span class="do">## We give extra space (5 lines) to the Left for your labels.</span></span>
<span id="cb6-142"><a href="#cb6-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-143"><a href="#cb6-143" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">5</span>, <span class="dv">2</span>, <span class="dv">2</span>))</span>
<span id="cb6-144"><a href="#cb6-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-145"><a href="#cb6-145" aria-hidden="true" tabindex="-1"></a><span class="do">## 3. Create the Image</span></span>
<span id="cb6-146"><a href="#cb6-146" aria-hidden="true" tabindex="-1"></a><span class="do">## We transform the matrix so it looks like the table (Row 1 at top)</span></span>
<span id="cb6-147"><a href="#cb6-147" aria-hidden="true" tabindex="-1"></a><span class="do">## t() transposes it, and we reverse the columns to flip the y-axis</span></span>
<span id="cb6-148"><a href="#cb6-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-149"><a href="#cb6-149" aria-hidden="true" tabindex="-1"></a>grid_data <span class="ot">&lt;-</span> <span class="fu">t</span>(fst_matrix[<span class="fu">nrow</span>(fst_matrix)<span class="sc">:</span><span class="dv">1</span>, ])</span>
<span id="cb6-150"><a href="#cb6-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-151"><a href="#cb6-151" aria-hidden="true" tabindex="-1"></a><span class="do">## Generate the heatmap colors</span></span>
<span id="cb6-152"><a href="#cb6-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-153"><a href="#cb6-153" aria-hidden="true" tabindex="-1"></a>my_colors <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">"#8ccddb"</span>, <span class="st">"#bf6c6c"</span>, <span class="st">"red"</span>))(<span class="dv">100</span>)</span>
<span id="cb6-154"><a href="#cb6-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-155"><a href="#cb6-155" aria-hidden="true" tabindex="-1"></a><span class="fu">image</span>(</span>
<span id="cb6-156"><a href="#cb6-156" aria-hidden="true" tabindex="-1"></a>  <span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>,           <span class="co"># X and Y coordinates</span></span>
<span id="cb6-157"><a href="#cb6-157" aria-hidden="true" tabindex="-1"></a>  grid_data,          <span class="co"># The data</span></span>
<span id="cb6-158"><a href="#cb6-158" aria-hidden="true" tabindex="-1"></a>  <span class="at">axes =</span> <span class="cn">FALSE</span>,       <span class="co"># Turn off default axes (we will draw our own)</span></span>
<span id="cb6-159"><a href="#cb6-159" aria-hidden="true" tabindex="-1"></a>  <span class="at">col =</span> my_colors,    <span class="co"># Colors</span></span>
<span id="cb6-160"><a href="#cb6-160" aria-hidden="true" tabindex="-1"></a>  <span class="at">xlab =</span> <span class="st">""</span>, <span class="at">ylab =</span> <span class="st">""</span></span>
<span id="cb6-161"><a href="#cb6-161" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-162"><a href="#cb6-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-163"><a href="#cb6-163" aria-hidden="true" tabindex="-1"></a><span class="do">## 4. Add Custom Axes</span></span>
<span id="cb6-164"><a href="#cb6-164" aria-hidden="true" tabindex="-1"></a><span class="do">## Axis 2 is the Left side. </span></span>
<span id="cb6-165"><a href="#cb6-165" aria-hidden="true" tabindex="-1"></a><span class="do">## We define positions 1:9, but label them in reverse</span></span>
<span id="cb6-166"><a href="#cb6-166" aria-hidden="true" tabindex="-1"></a><span class="do">## (Pop9 at bottom, Pop1 at top)</span></span>
<span id="cb6-167"><a href="#cb6-167" aria-hidden="true" tabindex="-1"></a><span class="fu">axis</span>(<span class="dv">2</span>, <span class="at">at =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>, <span class="at">labels =</span> <span class="fu">rev</span>(labs), <span class="at">las =</span> <span class="dv">2</span>,</span>
<span id="cb6-168"><a href="#cb6-168" aria-hidden="true" tabindex="-1"></a>     <span class="at">tick =</span> <span class="cn">FALSE</span>, <span class="at">cex.axis =</span> <span class="dv">1</span>)</span>
<span id="cb6-169"><a href="#cb6-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-170"><a href="#cb6-170" aria-hidden="true" tabindex="-1"></a><span class="do">## Axis 1 is the Bottom side.</span></span>
<span id="cb6-171"><a href="#cb6-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-172"><a href="#cb6-172" aria-hidden="true" tabindex="-1"></a><span class="fu">axis</span>(<span class="dv">1</span>, <span class="at">at =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>, <span class="at">labels =</span> labs, <span class="at">las =</span> <span class="dv">2</span>, <span class="at">tick =</span> <span class="cn">FALSE</span>,</span>
<span id="cb6-173"><a href="#cb6-173" aria-hidden="true" tabindex="-1"></a>     <span class="at">cex.axis =</span> <span class="dv">1</span>, <span class="at">las =</span> <span class="dv">1</span>)</span>
<span id="cb6-174"><a href="#cb6-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-175"><a href="#cb6-175" aria-hidden="true" tabindex="-1"></a><span class="do">## 5. Add Text Values</span></span>
<span id="cb6-176"><a href="#cb6-176" aria-hidden="true" tabindex="-1"></a><span class="do">## We loop through the original matrix to place the numbers</span></span>
<span id="cb6-177"><a href="#cb6-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-178"><a href="#cb6-178" aria-hidden="true" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="dv">9</span></span>
<span id="cb6-179"><a href="#cb6-179" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (row <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n) {</span>
<span id="cb6-180"><a href="#cb6-180" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (col <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n) {</span>
<span id="cb6-181"><a href="#cb6-181" aria-hidden="true" tabindex="-1"></a>    val <span class="ot">&lt;-</span> fst_matrix[row, col]</span>
<span id="cb6-182"><a href="#cb6-182" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb6-183"><a href="#cb6-183" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.na</span>(val)) {</span>
<span id="cb6-184"><a href="#cb6-184" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Logic: </span></span>
<span id="cb6-185"><a href="#cb6-185" aria-hidden="true" tabindex="-1"></a>      <span class="co"># x coordinate = column index</span></span>
<span id="cb6-186"><a href="#cb6-186" aria-hidden="true" tabindex="-1"></a>      <span class="co"># y coordinate = n + 1 - row index (because y=1 is the bottom)</span></span>
<span id="cb6-187"><a href="#cb6-187" aria-hidden="true" tabindex="-1"></a>      <span class="fu">text</span>(<span class="at">x =</span> col, <span class="at">y =</span> n <span class="sc">+</span> <span class="dv">1</span> <span class="sc">-</span> row, <span class="at">labels =</span> val, <span class="at">col =</span> <span class="st">"black"</span>, <span class="at">cex =</span> <span class="fl">0.9</span>)</span>
<span id="cb6-188"><a href="#cb6-188" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb6-189"><a href="#cb6-189" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb6-190"><a href="#cb6-190" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-191"><a href="#cb6-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-192"><a href="#cb6-192" aria-hidden="true" tabindex="-1"></a><span class="do">## --- 3. ADD LEGEND (TOP RIGHT) ---</span></span>
<span id="cb6-193"><a href="#cb6-193" aria-hidden="true" tabindex="-1"></a><span class="do">## We draw the legend in the empty white space (Coordinate x=6 to x=8 approx)</span></span>
<span id="cb6-194"><a href="#cb6-194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-195"><a href="#cb6-195" aria-hidden="true" tabindex="-1"></a><span class="do">## Legend settings</span></span>
<span id="cb6-196"><a href="#cb6-196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-197"><a href="#cb6-197" aria-hidden="true" tabindex="-1"></a>leg_x <span class="ot">&lt;-</span> <span class="fl">6.5</span>       <span class="co"># X starting position</span></span>
<span id="cb6-198"><a href="#cb6-198" aria-hidden="true" tabindex="-1"></a>leg_y <span class="ot">&lt;-</span> <span class="fl">6.0</span>       <span class="co"># Y starting position (bottom of legend)</span></span>
<span id="cb6-199"><a href="#cb6-199" aria-hidden="true" tabindex="-1"></a>leg_w <span class="ot">&lt;-</span> <span class="fl">0.5</span>       <span class="co"># Width of the bar</span></span>
<span id="cb6-200"><a href="#cb6-200" aria-hidden="true" tabindex="-1"></a>leg_h <span class="ot">&lt;-</span> <span class="fl">2.5</span>       <span class="co"># Height of the bar</span></span>
<span id="cb6-201"><a href="#cb6-201" aria-hidden="true" tabindex="-1"></a>num_cols <span class="ot">&lt;-</span> <span class="fu">length</span>(my_colors)</span>
<span id="cb6-202"><a href="#cb6-202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-203"><a href="#cb6-203" aria-hidden="true" tabindex="-1"></a><span class="do">## Draw the Gradient Bar</span></span>
<span id="cb6-204"><a href="#cb6-204" aria-hidden="true" tabindex="-1"></a><span class="do">## We stack tiny rectangles on top of each other</span></span>
<span id="cb6-205"><a href="#cb6-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-206"><a href="#cb6-206" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>num_cols) {</span>
<span id="cb6-207"><a href="#cb6-207" aria-hidden="true" tabindex="-1"></a>  y_start <span class="ot">&lt;-</span> leg_y <span class="sc">+</span> (i <span class="sc">-</span> <span class="dv">1</span>) <span class="sc">*</span> (leg_h <span class="sc">/</span> num_cols)</span>
<span id="cb6-208"><a href="#cb6-208" aria-hidden="true" tabindex="-1"></a>  y_end   <span class="ot">&lt;-</span> leg_y <span class="sc">+</span> (i) <span class="sc">*</span> (leg_h <span class="sc">/</span> num_cols)</span>
<span id="cb6-209"><a href="#cb6-209" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rect</span>(</span>
<span id="cb6-210"><a href="#cb6-210" aria-hidden="true" tabindex="-1"></a>    <span class="at">xleft =</span> leg_x, </span>
<span id="cb6-211"><a href="#cb6-211" aria-hidden="true" tabindex="-1"></a>    <span class="at">ybottom =</span> y_start, </span>
<span id="cb6-212"><a href="#cb6-212" aria-hidden="true" tabindex="-1"></a>    <span class="at">xright =</span> leg_x <span class="sc">+</span> leg_w, </span>
<span id="cb6-213"><a href="#cb6-213" aria-hidden="true" tabindex="-1"></a>    <span class="at">ytop =</span> y_end, </span>
<span id="cb6-214"><a href="#cb6-214" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> my_colors[i], </span>
<span id="cb6-215"><a href="#cb6-215" aria-hidden="true" tabindex="-1"></a>    <span class="at">border =</span> <span class="cn">NA</span></span>
<span id="cb6-216"><a href="#cb6-216" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb6-217"><a href="#cb6-217" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-218"><a href="#cb6-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-219"><a href="#cb6-219" aria-hidden="true" tabindex="-1"></a><span class="do">## Add a black border around the legend bar</span></span>
<span id="cb6-220"><a href="#cb6-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-221"><a href="#cb6-221" aria-hidden="true" tabindex="-1"></a><span class="fu">rect</span>(leg_x, leg_y, leg_x <span class="sc">+</span> leg_w, leg_y <span class="sc">+</span> leg_h, <span class="at">border =</span> <span class="st">"black"</span>)</span>
<span id="cb6-222"><a href="#cb6-222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-223"><a href="#cb6-223" aria-hidden="true" tabindex="-1"></a><span class="do">## Add Legend Text Labels (0, 0.5, 1.0)</span></span>
<span id="cb6-224"><a href="#cb6-224" aria-hidden="true" tabindex="-1"></a><span class="do">## We map the data range (0 to 0.83) to the legend height</span></span>
<span id="cb6-225"><a href="#cb6-225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-226"><a href="#cb6-226" aria-hidden="true" tabindex="-1"></a>min_val <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb6-227"><a href="#cb6-227" aria-hidden="true" tabindex="-1"></a>max_val <span class="ot">&lt;-</span> <span class="fu">max</span>(fst_matrix, <span class="at">na.rm=</span><span class="cn">TRUE</span>) <span class="co"># approx 0.83</span></span>
<span id="cb6-228"><a href="#cb6-228" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-229"><a href="#cb6-229" aria-hidden="true" tabindex="-1"></a><span class="do">## Label position 1 (Bottom)</span></span>
<span id="cb6-230"><a href="#cb6-230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-231"><a href="#cb6-231" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(<span class="at">x =</span> leg_x <span class="sc">+</span> leg_w <span class="sc">+</span> <span class="fl">0.2</span>, <span class="at">y =</span> leg_y, <span class="at">labels =</span> min_val,</span>
<span id="cb6-232"><a href="#cb6-232" aria-hidden="true" tabindex="-1"></a>     <span class="at">adj =</span> <span class="dv">0</span>, <span class="at">cex =</span> <span class="fl">0.8</span>)</span>
<span id="cb6-233"><a href="#cb6-233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-234"><a href="#cb6-234" aria-hidden="true" tabindex="-1"></a><span class="do">## Label position 2 (Middle)</span></span>
<span id="cb6-235"><a href="#cb6-235" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-236"><a href="#cb6-236" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(<span class="at">x =</span> leg_x <span class="sc">+</span> leg_w <span class="sc">+</span> <span class="fl">0.2</span>, <span class="at">y =</span> leg_y <span class="sc">+</span> (leg_h <span class="sc">/</span> <span class="dv">2</span>),</span>
<span id="cb6-237"><a href="#cb6-237" aria-hidden="true" tabindex="-1"></a>     <span class="at">labels =</span> <span class="fu">round</span>((max_val<span class="sc">/</span><span class="dv">2</span>), <span class="dv">2</span>), <span class="at">adj =</span> <span class="dv">0</span>, <span class="at">cex =</span> <span class="fl">0.8</span>)</span>
<span id="cb6-238"><a href="#cb6-238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-239"><a href="#cb6-239" aria-hidden="true" tabindex="-1"></a><span class="do">## Label position 3 (Top)</span></span>
<span id="cb6-240"><a href="#cb6-240" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(<span class="at">x =</span> leg_x <span class="sc">+</span> leg_w <span class="sc">+</span> <span class="fl">0.2</span>, <span class="at">y =</span> leg_y <span class="sc">+</span> leg_h,</span>
<span id="cb6-241"><a href="#cb6-241" aria-hidden="true" tabindex="-1"></a>     <span class="at">labels =</span> max_val, <span class="at">adj =</span> <span class="dv">0</span>, <span class="at">cex =</span> <span class="fl">0.8</span>)</span>
<span id="cb6-242"><a href="#cb6-242" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-243"><a href="#cb6-243" aria-hidden="true" tabindex="-1"></a><span class="do">## Add Title above legend</span></span>
<span id="cb6-244"><a href="#cb6-244" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-245"><a href="#cb6-245" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(<span class="at">x =</span> leg_x <span class="sc">+</span> (leg_w<span class="sc">/</span><span class="dv">2</span>), <span class="at">y =</span> leg_y <span class="sc">+</span> leg_h <span class="sc">+</span> <span class="fl">0.3</span>,</span>
<span id="cb6-246"><a href="#cb6-246" aria-hidden="true" tabindex="-1"></a>     <span class="at">labels =</span> <span class="fu">expression</span>(<span class="fu">italic</span>(F[st])), <span class="at">font =</span> <span class="dv">2</span>, <span class="at">cex =</span> <span class="dv">1</span>)</span>
<span id="cb6-247"><a href="#cb6-247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-248"><a href="#cb6-248" aria-hidden="true" tabindex="-1"></a><span class="do">## --- 4. ADD FIGURE LABEL "B" (Top Left) ---</span></span>
<span id="cb6-249"><a href="#cb6-249" aria-hidden="true" tabindex="-1"></a><span class="do">## side=3 (Top), line=1 (distance from plot), adj implies</span></span>
<span id="cb6-250"><a href="#cb6-250" aria-hidden="true" tabindex="-1"></a><span class="do">## alignment (-0.1 moves it into left margin)</span></span>
<span id="cb6-251"><a href="#cb6-251" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-252"><a href="#cb6-252" aria-hidden="true" tabindex="-1"></a><span class="fu">mtext</span>(<span class="st">"B"</span>, <span class="at">side =</span> <span class="dv">3</span>, <span class="at">at =</span> <span class="sc">-</span><span class="fl">0.48</span>, <span class="at">line =</span> <span class="dv">1</span>, <span class="at">adj =</span> <span class="sc">-</span><span class="fl">0.15</span>,</span>
<span id="cb6-253"><a href="#cb6-253" aria-hidden="true" tabindex="-1"></a>      <span class="at">cex =</span> <span class="dv">1</span>, <span class="at">font =</span> <span class="dv">1</span>)</span>
<span id="cb6-254"><a href="#cb6-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-255"><a href="#cb6-255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-256"><a href="#cb6-256" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"/Users/dpadil10/ASU Dropbox/Dylan Padilla/Yale/aRanSyl-PopGen/data/aRanSyl_genind_pops.RData"</span>)</span>
<span id="cb6-257"><a href="#cb6-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-258"><a href="#cb6-258" aria-hidden="true" tabindex="-1"></a><span class="fu">indNames</span>(aRanSyl_genind_pops)</span>
<span id="cb6-259"><a href="#cb6-259" aria-hidden="true" tabindex="-1"></a>aRanSyl_genind_pops<span class="sc">$</span>pop</span>
<span id="cb6-260"><a href="#cb6-260" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-261"><a href="#cb6-261" aria-hidden="true" tabindex="-1"></a>tab <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">id =</span> <span class="fu">indNames</span>(aRanSyl_genind_pops), <span class="at">pop =</span> aRanSyl_genind_pops<span class="sc">$</span>pop)</span>
<span id="cb6-262"><a href="#cb6-262" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(tab)</span>
<span id="cb6-263"><a href="#cb6-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-264"><a href="#cb6-264" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(meta)</span>
<span id="cb6-265"><a href="#cb6-265" aria-hidden="true" tabindex="-1"></a>admix <span class="ot">&lt;-</span> <span class="fu">merge</span>(meta, tab[<span class="sc">-</span><span class="dv">2</span>], <span class="at">by =</span> <span class="st">"id"</span>)</span>
<span id="cb6-266"><a href="#cb6-266" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(admix)</span>
<span id="cb6-267"><a href="#cb6-267" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-268"><a href="#cb6-268" aria-hidden="true" tabindex="-1"></a>pop_admix <span class="ot">&lt;-</span> admix[ , <span class="fu">c</span>(<span class="st">"V1"</span>, <span class="st">"V2"</span>, <span class="st">"V3"</span>, <span class="st">"V4"</span>, <span class="st">"V5"</span>, <span class="st">"V6"</span>, <span class="st">"V7"</span>, <span class="st">"V8"</span>,</span>
<span id="cb6-269"><a href="#cb6-269" aria-hidden="true" tabindex="-1"></a><span class="st">"V9"</span>)]</span>
<span id="cb6-270"><a href="#cb6-270" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(pop_admix)</span>
<span id="cb6-271"><a href="#cb6-271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-272"><a href="#cb6-272" aria-hidden="true" tabindex="-1"></a><span class="do">## find the column name with the highest value</span></span>
<span id="cb6-273"><a href="#cb6-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-274"><a href="#cb6-274" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb6-275"><a href="#cb6-275" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(pop_admix)) {</span>
<span id="cb6-276"><a href="#cb6-276" aria-hidden="true" tabindex="-1"></a>    max_col_name <span class="ot">&lt;-</span> <span class="fu">names</span>(pop_admix[i, ])[<span class="fu">apply</span>(pop_admix[i, ],</span>
<span id="cb6-277"><a href="#cb6-277" aria-hidden="true" tabindex="-1"></a>                                                <span class="dv">1</span>, which.max)[<span class="fu">which.max</span>(<span class="fu">apply</span>(pop_admix[i, ],</span>
<span id="cb6-278"><a href="#cb6-278" aria-hidden="true" tabindex="-1"></a>                                                                              <span class="dv">1</span>, max))]]</span>
<span id="cb6-279"><a href="#cb6-279" aria-hidden="true" tabindex="-1"></a>    obj <span class="ot">&lt;-</span> <span class="fu">c</span>(obj, max_col_name)</span>
<span id="cb6-280"><a href="#cb6-280" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-281"><a href="#cb6-281" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-282"><a href="#cb6-282" aria-hidden="true" tabindex="-1"></a>obj</span>
<span id="cb6-283"><a href="#cb6-283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-284"><a href="#cb6-284" aria-hidden="true" tabindex="-1"></a><span class="do">## adding a column to the admix dataframe with the new vector of names</span></span>
<span id="cb6-285"><a href="#cb6-285" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-286"><a href="#cb6-286" aria-hidden="true" tabindex="-1"></a>admix<span class="sc">$</span>pop <span class="ot">&lt;-</span> obj</span>
<span id="cb6-287"><a href="#cb6-287" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(admix)</span>
<span id="cb6-288"><a href="#cb6-288" aria-hidden="true" tabindex="-1"></a>admix<span class="sc">$</span>pop <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(admix<span class="sc">$</span>pop)</span>
<span id="cb6-289"><a href="#cb6-289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-290"><a href="#cb6-290" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(admix, <span class="st">"/gpfs/gibbs/project/skelly/dp996/Wood-frog_ddRADseq/data/aRanSyl_metadata.csv"</span>, <span class="at">sep =</span> <span class="st">','</span>)</span>
<span id="cb6-291"><a href="#cb6-291" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-292"><a href="#cb6-292" aria-hidden="true" tabindex="-1"></a>aRanSyl_genind<span class="sc">$</span>pop <span class="ot">&lt;-</span> admix<span class="sc">$</span>pop</span>
<span id="cb6-293"><a href="#cb6-293" aria-hidden="true" tabindex="-1"></a>aRanSyl_genind_pops <span class="ot">&lt;-</span> aRanSyl_genind</span>
<span id="cb6-294"><a href="#cb6-294" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(aRanSyl_genind_pops<span class="sc">$</span>pop)</span>
<span id="cb6-295"><a href="#cb6-295" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(aRanSyl_genind_pops, <span class="at">file =</span> <span class="st">"/gpfs/gibbs/project/skelly/dp996/Wood-frog_ddRADseq/data/aRanSyl_genind_pops.RData"</span>)</span>
<span id="cb6-296"><a href="#cb6-296" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-297"><a href="#cb6-297" aria-hidden="true" tabindex="-1"></a>aRanSyl_gl_pops <span class="ot">&lt;-</span> <span class="fu">gi2gl</span>(aRanSyl_genind, <span class="at">parallel =</span> <span class="cn">FALSE</span>, <span class="at">verbose =</span> <span class="cn">NULL</span>)</span>
<span id="cb6-298"><a href="#cb6-298" aria-hidden="true" tabindex="-1"></a>aRanSyl_gl_pops</span>
<span id="cb6-299"><a href="#cb6-299" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(aRanSyl_gl_pops, <span class="at">file =</span> <span class="st">"/gpfs/gibbs/project/skelly/dp996/Wood-frog_ddRADseq/data/aRanSyl_gl_pops.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><img src="imgs/aRanSyl_structure.png" class="img-fluid"></p>
<p><strong>This the script to run the demographic history inference</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="co">#| output: true</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> ./scripts/SFS.sh</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>#!/bin/bash

#SBATCH --partition=day                    # which partition to run the job, options are in the Amarel guide
#SBATCH --job-name=aRanSyl_SFS                       # job name for listing in queue
#SBATCH --output=/Users/dpadil10/ASU Dropbox/Dylan Padilla/Yale/aRanSyl-PopGen/results/logs/SFS/slurm-%j-%x.out
#SBATCH --error=/Users/dpadil10/ASU Dropbox/Dylan Padilla/Yale/aRanSyl-PopGen/results/logs/SFS/slurm-%j-%x.err
#SBATCH --mem=100G                               # memory to allocate in Mb
#SBATCH -n 10                                   # number of cores to use
#SBATCH --time=1-00:00:00                       # maximum run time days-hours:minutes:seconds
#SBATCH --mail-user=dylan.padilla@yale.edu           # email address to send status updates
#SBATCH --mail-type=FAIL,END                     # email for the following reasons

module --force purge
conda activate easySFS

## we will work with a subset of sites that are already filtered for low missing data proportions:

VCF="/Users/dpadil10/ASU Dropbox/Dylan Padilla/Yale/aRanSyl-PopGen/data/aRanSyl_demographic_0.5pruned.vcf.gz"
POPS="/Users/dpadil10/ASU Dropbox/Dylan Padilla/Yale/aRanSyl-PopGen/data/pops_demographic_file.txt"

## setting directory

cd /Users/dpadil10/ASU Dropbox/Dylan Padilla/Yale/aRanSyl-PopGen/SFS/aRanSyl_SFS/

## estimate projections

/Users/dpadil10/ASU Dropbox/Dylan Padilla/Yale/aRanSyl-PopGen/SFS/easySFS/easySFS.py -i $VCF -p $POPS --preview -a

## running easySFS.py based on the projections recommended

/Users/dpadil10/ASU Dropbox/Dylan Padilla/Yale/aRanSyl-PopGen/SFS/easySFS/easySFS.py -i $VCF -p $POPS -a --proj 50,32</code></pre>
</div>
</div>
<p><strong>Here are results of the script described above:</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="co">#| output: true</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="ex">tree</span> ./SFS/aRanSyl_SFS</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>./SFS/aRanSyl_SFS
 output
     dadi
    &nbsp;&nbsp;  pop1-32.sfs
    &nbsp;&nbsp;  pop9-50.sfs
    &nbsp;&nbsp;  pop9-pop1.sfs
     datadict.txt
     fastsimcoal2
         aRanSyl_demographic_0_jointMAFpop1_0.obs
         aRanSyl_demographic_0_MSFS.obs
         pop1_MAFpop0.obs
         pop9_MAFpop0.obs
         proj.txt

4 directories, 9 files</code></pre>
</div>
</div>
<p><strong>fastsimcoal2 runs</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co">#| output: true</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="ex">tree</span> ./SFS/fastsimcoal</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>./SFS/fastsimcoal
 demographic_model_AIC_table.txt
 diff_geneflow
&nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp;  diff_geneflow_MSFS.obs
&nbsp;&nbsp;  diff_geneflow.est
&nbsp;&nbsp;  diff_geneflow.tpl
&nbsp;&nbsp;  fastsimcoal2_output.log
&nbsp;&nbsp;  fastsimcoal2.sh
&nbsp;&nbsp;  fsc-selection
&nbsp;&nbsp; &nbsp;&nbsp;  bestrun
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  calculateAIC.sh
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.AIC
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  plotModel.r
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  SFStools.R
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  fsc-selectbestrun.sh
&nbsp;&nbsp; &nbsp;&nbsp;  run1
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run10
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run11
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run12
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run13
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run14
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run15
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run16
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run17
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run18
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run19
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run2
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run20
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run3
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run4
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run5
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run6
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run7
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run8
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run9
&nbsp;&nbsp; &nbsp;&nbsp;      diff_geneflow.bestlhoods
&nbsp;&nbsp;  pop1_MAFpop0.obs
&nbsp;&nbsp;  pop9_MAFpop0.obs
&nbsp;&nbsp;  run1
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run10
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run11
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run12
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run13
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run14
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run15
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run16
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run17
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run18
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run19
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run2
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run20
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run3
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run4
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run5
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run6
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run7
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run8
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  diff_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run9
&nbsp;&nbsp;      diff_geneflow
&nbsp;&nbsp;     &nbsp;&nbsp;  diff_geneflow_1.simparam
&nbsp;&nbsp;     &nbsp;&nbsp;  diff_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp;     &nbsp;&nbsp;  diff_geneflow_maxL.par
&nbsp;&nbsp;     &nbsp;&nbsp;  diff_geneflow.bestlhoods
&nbsp;&nbsp;     &nbsp;&nbsp;  diff_geneflow.brent_lhoods
&nbsp;&nbsp;     &nbsp;&nbsp;  diff_geneflow.pv
&nbsp;&nbsp;      diff_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp;      diff_geneflow.est
&nbsp;&nbsp;      diff_geneflow.par
&nbsp;&nbsp;      diff_geneflow.tpl
&nbsp;&nbsp;      seed.txt
 early_geneflow
&nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp;  early_geneflow_MSFS.obs
&nbsp;&nbsp;  early_geneflow.est
&nbsp;&nbsp;  early_geneflow.tpl
&nbsp;&nbsp;  fastsimcoal2_output.log
&nbsp;&nbsp;  fastsimcoal2.sh
&nbsp;&nbsp;  fsc-selection
&nbsp;&nbsp; &nbsp;&nbsp;  bestrun
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  calculateAIC.sh
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.AIC
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  fsc-selectbestrun.sh
&nbsp;&nbsp; &nbsp;&nbsp;  run1
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run10
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run11
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run12
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run13
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run14
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run15
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run16
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run17
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run18
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run19
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run2
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run20
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run3
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run4
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run5
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run6
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run7
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run8
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run9
&nbsp;&nbsp; &nbsp;&nbsp;      early_geneflow.bestlhoods
&nbsp;&nbsp;  pop1_MAFpop0.obs
&nbsp;&nbsp;  pop9_MAFpop0.obs
&nbsp;&nbsp;  run1
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run10
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run11
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run12
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run13
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run14
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run15
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run16
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run17
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run18
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run19
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run2
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run20
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run3
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run4
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run5
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run6
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run7
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run8
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  early_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run9
&nbsp;&nbsp;      early_geneflow
&nbsp;&nbsp;     &nbsp;&nbsp;  early_geneflow_1.simparam
&nbsp;&nbsp;     &nbsp;&nbsp;  early_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp;     &nbsp;&nbsp;  early_geneflow_maxL.par
&nbsp;&nbsp;     &nbsp;&nbsp;  early_geneflow.bestlhoods
&nbsp;&nbsp;     &nbsp;&nbsp;  early_geneflow.brent_lhoods
&nbsp;&nbsp;     &nbsp;&nbsp;  early_geneflow.pv
&nbsp;&nbsp;      early_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp;      early_geneflow.est
&nbsp;&nbsp;      early_geneflow.par
&nbsp;&nbsp;      early_geneflow.tpl
&nbsp;&nbsp;      seed.txt
 no_geneflow
&nbsp;&nbsp;  fastsimcoal2_output.log
&nbsp;&nbsp;  fastsimcoal2.sh
&nbsp;&nbsp;  fsc-selection
&nbsp;&nbsp; &nbsp;&nbsp;  bestrun
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  calculateAIC.sh
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_2DMAFs.pdf
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_model.pdf
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_MSFS.obs
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.AIC
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  plotModel.r
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  plotModel.r~
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  SFStools.R
&nbsp;&nbsp; &nbsp;&nbsp;  fsc-selectbestrun.sh
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  run1
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run12
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run14
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run15
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run16
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run17
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run18
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run19
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run2
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run20
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run4
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run5
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run6
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run8
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run9
&nbsp;&nbsp; &nbsp;&nbsp;      no_geneflow.bestlhoods
&nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp;  no_geneflow_MSFS.obs
&nbsp;&nbsp;  no_geneflow.est
&nbsp;&nbsp;  no_geneflow.tpl
&nbsp;&nbsp;  pop1_MAFpop0.obs
&nbsp;&nbsp;  pop9_MAFpop0.obs
&nbsp;&nbsp;  run1
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run10
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run11
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run12
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run13
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run14
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run15
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run16
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run17
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run18
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run19
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run2
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run20
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run3
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run4
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run5
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run6
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run7
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run8
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  no_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run9
&nbsp;&nbsp;      no_geneflow
&nbsp;&nbsp;     &nbsp;&nbsp;  no_geneflow_1.simparam
&nbsp;&nbsp;     &nbsp;&nbsp;  no_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp;     &nbsp;&nbsp;  no_geneflow_maxL.par
&nbsp;&nbsp;     &nbsp;&nbsp;  no_geneflow.bestlhoods
&nbsp;&nbsp;     &nbsp;&nbsp;  no_geneflow.brent_lhoods
&nbsp;&nbsp;     &nbsp;&nbsp;  no_geneflow.pv
&nbsp;&nbsp;      no_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp;      no_geneflow.est
&nbsp;&nbsp;      no_geneflow.par
&nbsp;&nbsp;      no_geneflow.tpl
&nbsp;&nbsp;      seed.txt
 ongoing_geneflow
&nbsp;&nbsp;  fastsimcoal2.sh
&nbsp;&nbsp;  fsc-selection
&nbsp;&nbsp; &nbsp;&nbsp;  bestrun
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  calculateAIC.sh
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.AIC
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  fsc-selectbestrun.sh
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  run1
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run10
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run11
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run12
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run13
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run14
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run15
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run16
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run17
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run18
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run19
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run2
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run20
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run3
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run4
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run5
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run6
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run7
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run8
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp;  run9
&nbsp;&nbsp; &nbsp;&nbsp;      ongoing_geneflow.bestlhoods
&nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp;  ongoing_geneflow_MSFS.obs
&nbsp;&nbsp;  ongoing_geneflow.est
&nbsp;&nbsp;  ongoing_geneflow.tpl
&nbsp;&nbsp;  pop1_MAFpop0.obs
&nbsp;&nbsp;  pop9_MAFpop0.obs
&nbsp;&nbsp;  run1
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run10
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run11
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run12
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run13
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run14
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run15
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run16
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run17
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run18
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run19
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run2
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run20
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run3
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run4
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run5
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run6
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run7
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run8
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_1.simparam
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_maxL.par
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.brent_lhoods
&nbsp;&nbsp; &nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.pv
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.est
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.par
&nbsp;&nbsp; &nbsp;&nbsp;  ongoing_geneflow.tpl
&nbsp;&nbsp; &nbsp;&nbsp;  seed.txt
&nbsp;&nbsp;  run9
&nbsp;&nbsp;      ongoing_geneflow
&nbsp;&nbsp;     &nbsp;&nbsp;  ongoing_geneflow_1.simparam
&nbsp;&nbsp;     &nbsp;&nbsp;  ongoing_geneflow_jointMAFpop1_0.txt
&nbsp;&nbsp;     &nbsp;&nbsp;  ongoing_geneflow_maxL.par
&nbsp;&nbsp;     &nbsp;&nbsp;  ongoing_geneflow.bestlhoods
&nbsp;&nbsp;     &nbsp;&nbsp;  ongoing_geneflow.brent_lhoods
&nbsp;&nbsp;     &nbsp;&nbsp;  ongoing_geneflow.pv
&nbsp;&nbsp;      ongoing_geneflow_jointMAFpop1_0.obs
&nbsp;&nbsp;      ongoing_geneflow.est
&nbsp;&nbsp;      ongoing_geneflow.par
&nbsp;&nbsp;      ongoing_geneflow.tpl
&nbsp;&nbsp;      seed.txt
 recent_geneflow
     fastsimcoal2_output.log
     fastsimcoal2.sh
     fsc-selection
    &nbsp;&nbsp;  bestrun
    &nbsp;&nbsp; &nbsp;&nbsp;  calculateAIC.sh
    &nbsp;&nbsp; &nbsp;&nbsp;  plotModel.r
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_2DMAFs.png
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.obs
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_maxL.par
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_model.pdf
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_model.R
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_model.R~
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_MSFS.obs
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.AIC
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.est
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.tpl
    &nbsp;&nbsp; &nbsp;&nbsp;  SFStools.R
    &nbsp;&nbsp; &nbsp;&nbsp;  SFStools.R~
    &nbsp;&nbsp;  fsc-selectbestrun.sh
    &nbsp;&nbsp;  recent_geneflow.tpl
    &nbsp;&nbsp;  run1
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp;  run10
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp;  run11
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp;  run12
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp;  run13
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp;  run14
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp;  run15
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp;  run16
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp;  run17
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp;  run18
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp;  run19
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp;  run2
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp;  run20
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp;  run3
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp;  run4
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp;  run5
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp;  run6
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp;  run7
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp;  run8
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp;  run9
    &nbsp;&nbsp;      recent_geneflow.bestlhoods
     pop1_MAFpop0.obs
     pop9_MAFpop0.obs
     recent_geneflow_jointMAFpop1_0.obs
     recent_geneflow_MSFS.obs
     recent_geneflow.est
     recent_geneflow.tpl
     run1
    &nbsp;&nbsp;  recent_geneflow
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_1.simparam
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.txt
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_maxL.par
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.brent_lhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.pv
    &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.obs
    &nbsp;&nbsp;  recent_geneflow.est
    &nbsp;&nbsp;  recent_geneflow.par
    &nbsp;&nbsp;  recent_geneflow.tpl
    &nbsp;&nbsp;  seed.txt
     run10
    &nbsp;&nbsp;  recent_geneflow
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_1.simparam
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.txt
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_maxL.par
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.brent_lhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.pv
    &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.obs
    &nbsp;&nbsp;  recent_geneflow.est
    &nbsp;&nbsp;  recent_geneflow.par
    &nbsp;&nbsp;  recent_geneflow.tpl
    &nbsp;&nbsp;  seed.txt
     run11
    &nbsp;&nbsp;  recent_geneflow
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_1.simparam
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.txt
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_maxL.par
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.brent_lhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.pv
    &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.obs
    &nbsp;&nbsp;  recent_geneflow.est
    &nbsp;&nbsp;  recent_geneflow.par
    &nbsp;&nbsp;  recent_geneflow.tpl
    &nbsp;&nbsp;  seed.txt
     run12
    &nbsp;&nbsp;  recent_geneflow
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_1.simparam
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.txt
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_maxL.par
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.brent_lhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.pv
    &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.obs
    &nbsp;&nbsp;  recent_geneflow.est
    &nbsp;&nbsp;  recent_geneflow.par
    &nbsp;&nbsp;  recent_geneflow.tpl
    &nbsp;&nbsp;  seed.txt
     run13
    &nbsp;&nbsp;  recent_geneflow
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_1.simparam
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.txt
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_maxL.par
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.brent_lhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.pv
    &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.obs
    &nbsp;&nbsp;  recent_geneflow.est
    &nbsp;&nbsp;  recent_geneflow.par
    &nbsp;&nbsp;  recent_geneflow.tpl
    &nbsp;&nbsp;  seed.txt
     run14
    &nbsp;&nbsp;  recent_geneflow
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_1.simparam
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.txt
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_maxL.par
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.brent_lhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.pv
    &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.obs
    &nbsp;&nbsp;  recent_geneflow.est
    &nbsp;&nbsp;  recent_geneflow.par
    &nbsp;&nbsp;  recent_geneflow.tpl
    &nbsp;&nbsp;  seed.txt
     run15
    &nbsp;&nbsp;  recent_geneflow
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_1.simparam
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.txt
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_maxL.par
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.brent_lhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.pv
    &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.obs
    &nbsp;&nbsp;  recent_geneflow.est
    &nbsp;&nbsp;  recent_geneflow.par
    &nbsp;&nbsp;  recent_geneflow.tpl
    &nbsp;&nbsp;  seed.txt
     run16
    &nbsp;&nbsp;  recent_geneflow
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_1.simparam
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.txt
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_maxL.par
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.brent_lhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.pv
    &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.obs
    &nbsp;&nbsp;  recent_geneflow.est
    &nbsp;&nbsp;  recent_geneflow.par
    &nbsp;&nbsp;  recent_geneflow.tpl
    &nbsp;&nbsp;  seed.txt
     run17
    &nbsp;&nbsp;  recent_geneflow
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_1.simparam
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.txt
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_maxL.par
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.brent_lhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.pv
    &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.obs
    &nbsp;&nbsp;  recent_geneflow.est
    &nbsp;&nbsp;  recent_geneflow.par
    &nbsp;&nbsp;  recent_geneflow.tpl
    &nbsp;&nbsp;  seed.txt
     run18
    &nbsp;&nbsp;  recent_geneflow
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_1.simparam
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.txt
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_maxL.par
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.brent_lhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.pv
    &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.obs
    &nbsp;&nbsp;  recent_geneflow.est
    &nbsp;&nbsp;  recent_geneflow.par
    &nbsp;&nbsp;  recent_geneflow.tpl
    &nbsp;&nbsp;  seed.txt
     run19
    &nbsp;&nbsp;  recent_geneflow
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_1.simparam
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.txt
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_maxL.par
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.brent_lhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.pv
    &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.obs
    &nbsp;&nbsp;  recent_geneflow.est
    &nbsp;&nbsp;  recent_geneflow.par
    &nbsp;&nbsp;  recent_geneflow.tpl
    &nbsp;&nbsp;  seed.txt
     run2
    &nbsp;&nbsp;  recent_geneflow
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_1.simparam
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.txt
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_maxL.par
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.brent_lhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.pv
    &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.obs
    &nbsp;&nbsp;  recent_geneflow.est
    &nbsp;&nbsp;  recent_geneflow.par
    &nbsp;&nbsp;  recent_geneflow.tpl
    &nbsp;&nbsp;  seed.txt
     run20
    &nbsp;&nbsp;  recent_geneflow
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_1.simparam
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.txt
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_maxL.par
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.brent_lhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.pv
    &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.obs
    &nbsp;&nbsp;  recent_geneflow.est
    &nbsp;&nbsp;  recent_geneflow.par
    &nbsp;&nbsp;  recent_geneflow.tpl
    &nbsp;&nbsp;  seed.txt
     run3
    &nbsp;&nbsp;  recent_geneflow
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_1.simparam
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.txt
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_maxL.par
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.brent_lhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.pv
    &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.obs
    &nbsp;&nbsp;  recent_geneflow.est
    &nbsp;&nbsp;  recent_geneflow.par
    &nbsp;&nbsp;  recent_geneflow.tpl
    &nbsp;&nbsp;  seed.txt
     run4
    &nbsp;&nbsp;  recent_geneflow
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_1.simparam
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.txt
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_maxL.par
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.brent_lhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.pv
    &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.obs
    &nbsp;&nbsp;  recent_geneflow.est
    &nbsp;&nbsp;  recent_geneflow.par
    &nbsp;&nbsp;  recent_geneflow.tpl
    &nbsp;&nbsp;  seed.txt
     run5
    &nbsp;&nbsp;  recent_geneflow
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_1.simparam
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.txt
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_maxL.par
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.brent_lhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.pv
    &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.obs
    &nbsp;&nbsp;  recent_geneflow.est
    &nbsp;&nbsp;  recent_geneflow.par
    &nbsp;&nbsp;  recent_geneflow.tpl
    &nbsp;&nbsp;  seed.txt
     run6
    &nbsp;&nbsp;  recent_geneflow
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_1.simparam
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.txt
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_maxL.par
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.brent_lhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.pv
    &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.obs
    &nbsp;&nbsp;  recent_geneflow.est
    &nbsp;&nbsp;  recent_geneflow.par
    &nbsp;&nbsp;  recent_geneflow.tpl
    &nbsp;&nbsp;  seed.txt
     run7
    &nbsp;&nbsp;  recent_geneflow
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_1.simparam
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.txt
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_maxL.par
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.brent_lhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.pv
    &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.obs
    &nbsp;&nbsp;  recent_geneflow.est
    &nbsp;&nbsp;  recent_geneflow.par
    &nbsp;&nbsp;  recent_geneflow.tpl
    &nbsp;&nbsp;  seed.txt
     run8
    &nbsp;&nbsp;  recent_geneflow
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_1.simparam
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.txt
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow_maxL.par
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.bestlhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.brent_lhoods
    &nbsp;&nbsp; &nbsp;&nbsp;  recent_geneflow.pv
    &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.obs
    &nbsp;&nbsp;  recent_geneflow.est
    &nbsp;&nbsp;  recent_geneflow.par
    &nbsp;&nbsp;  recent_geneflow.tpl
    &nbsp;&nbsp;  seed.txt
     run9
         recent_geneflow
        &nbsp;&nbsp;  recent_geneflow_1.simparam
        &nbsp;&nbsp;  recent_geneflow_jointMAFpop1_0.txt
        &nbsp;&nbsp;  recent_geneflow_maxL.par
        &nbsp;&nbsp;  recent_geneflow.bestlhoods
        &nbsp;&nbsp;  recent_geneflow.brent_lhoods
        &nbsp;&nbsp;  recent_geneflow.pv
         recent_geneflow_jointMAFpop1_0.obs
         recent_geneflow.est
         recent_geneflow.par
         recent_geneflow.tpl
         seed.txt

311 directories, 1276 files</code></pre>
</div>
</div>
<p><strong>According to the AIC values, a recent gene flow model was strongly supported for inferences</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="co">#| output: true</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> ./SFS/fastsimcoal/demographic_model_AIC_table.txt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>deltaL  AIC
3266.32400000002    645305.776535448
deltaL  AIC
3423.82200000001    646027.081629401
deltaL  AIC
6016.77800000002    657964.08529418
deltaL  AIC
3399.46900000001    645912.931919862
deltaL  AIC
3235.62600000002    645160.407021079</code></pre>
</div>
</div>
<p><strong>Here are the .tpl and .est files of the model:</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="co">#| output: true</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> ./SFS/fastsimcoal/recent_geneflow/fsc-selection/bestrun/recent_geneflow.tpl</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>//Parameters for the coalescence simulation program : fastsimcoal.exe
2 samples to simulate :
//Population effective sizes (number of genes)
$NPOP9$
$NPOP1$
//Sample sizes and age
50
32
//Growth rates: negative growth implies population expansion
0
0
//Number of migration matrices : 0 implies no migration between demes
2
//Migration matrix 0
0   $MIG1$      
$MIG2$  0
//Migration matrix 1
0       0
0   0
//historical event: time, source, sink, migrants, new deme size, new growth rate, migration matrix index
2 historical event
$CHANGM$ 0 0 1 1 0 1
$TDIV$ 0 1 1 $RSANC$ 0 1
//Number of independent loci [chromosome] 
1 0
//Per chromosome: Number of contiguous linkage Block: a block is a set of contiguous loci
1
//per Block:data type, number of loci, per generation recombination and mutation rates and optional parameters
FREQ 1 0 1.5e-9 OUTEXP</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="co">#| output: true</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> ./SFS/fastsimcoal/recent_geneflow/fsc-selection/bestrun/recent_geneflow.est</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>// Priors and rules file
// *********************

[PARAMETERS]
//#isInt? #name   #dist.#min  #max
//all Ns are in number of haploid individuals
1  $NPOP9$      logunif 40   100000  output
1  $NPOP1$      logunif 40   100000  output
0  $RSANC$          logunif 0.1  100     output
0  $MIG1$               logunif 0.000001 0.001   output
0  $MIG2$               logunif 0.000001 0.001   output
0  $TPROP$          logunif 0.0001   0.5     output
1  $TDIV$           logunif 100  100000  output

[COMPLEX PARAMETERS]
1  $CHANGM$ = $TDIV$ * $TPROP$       output</code></pre>
</div>
</div>
<p><strong>Here are the results of the model:</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="co">#| output: true</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-la</span> ./SFS/fastsimcoal/recent_geneflow/fsc-selection/bestrun/</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>total 552
drwxr-xr-x@ 18 dpadil10  staff     576 Jan 28 18:07 .
drwxr-xr-x@ 25 dpadil10  staff     800 Jan 20 23:24 ..
-rw-r--r--@  1 dpadil10  staff      12 Jan 28 18:07 .Rhistory
-rw-r--r--@  1 dpadil10  staff    1668 Jan 17 01:18 calculateAIC.sh
-rw-r--r--@  1 dpadil10  staff   11423 Jan 20 21:57 plotModel.r
-rw-r--r--@  1 dpadil10  staff  148481 Jan 21 11:14 recent_geneflow_2DMAFs.png
-rw-r--r--@  1 dpadil10  staff   17507 Jan 20 21:48 recent_geneflow_jointMAFpop1_0.obs
-rw-r--r--@  1 dpadil10  staff     877 Jan 20 21:48 recent_geneflow_maxL.par
-rw-r--r--@  1 dpadil10  staff    5515 Jan 20 21:58 recent_geneflow_model.pdf
-rw-r--r--@  1 dpadil10  staff    6032 Jan 28 13:54 recent_geneflow_model.R
-rw-r--r--@  1 dpadil10  staff    6026 Jan 21 11:16 recent_geneflow_model.R~
-rw-r--r--@  1 dpadil10  staff   17081 Jan 20 22:59 recent_geneflow_MSFS.obs
-rw-r--r--@  1 dpadil10  staff      45 Jan 20 21:49 recent_geneflow.AIC
-rw-r--r--@  1 dpadil10  staff     174 Jan 20 21:48 recent_geneflow.bestlhoods
-rw-r--r--@  1 dpadil10  staff     533 Jan 20 21:48 recent_geneflow.est
-rw-r--r--@  1 dpadil10  staff     845 Jan 20 21:48 recent_geneflow.tpl
-rw-r--r--@  1 dpadil10  staff   11707 Jan 21 11:14 SFStools.R
-rw-r--r--@  1 dpadil10  staff   11707 Jan 21 11:09 SFStools.R~</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="co">#| output: true</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> ./SFS/fastsimcoal/recent_geneflow/fsc-selection/bestrun/recent_geneflow_maxL.par</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>//Parameters for the coalescence simulation program : fastsimcoal.exe
2 samples to simulate :
//Population effective sizes (number of genes)
1162
13188
//Sample sizes and age
50
32
//Growth rates: negative growth implies population expansion
0
0
//Number of migration matrices : 0 implies no migration between demes
2
//Migration matrix 0
0   8.070451464149667e-05       
5.329187459143579e-05  0
//Migration matrix 1
0       0
0   0
//historical event: time, source, sink, migrants, new deme size, new growth rate, migration matrix index
2 historical event
442 0 0 1 1 0 1
3313 0 1 1 1.436895375456696e-01 0 1
//Number of independent loci [chromosome] 
1 0
//Per chromosome: Number of contiguous linkage Block: a block is a set of contiguous loci
1
//per Block:data type, number of loci, per generation recombination and mutation rates and optional parameters
FREQ 1 0 1.5e-9 OUTEXP</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="co">#| output: true</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> ./SFS/fastsimcoal/recent_geneflow/fsc-selection/bestrun/recent_geneflow.bestlhoods</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$NPOP9$ $NPOP1$ $RSANC$ $MIG1$  $MIG2$  $TPROP$ $TDIV$  $CHANGM$    MaxEstLhood MaxObsLhood
1162    13188   0.1436895   8.07045e-05 5.32919e-05 0.1333612   3313    442 -140091.328 -136855.702</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">layout</span>(<span class="fu">matrix</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">0</span>,</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>                <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">0</span>,</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>                <span class="dv">0</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">0</span>,</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>                <span class="dv">0</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">0</span>), <span class="at">ncol =</span> <span class="dv">4</span>, <span class="at">nrow =</span> <span class="dv">4</span>, <span class="at">byrow =</span> <span class="cn">TRUE</span>))</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a><span class="do">##layout.show(n = 2)</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a><span class="do">## --- 1. Define Model Parameters ---</span></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>t_div   <span class="ot">&lt;-</span> <span class="dv">3313</span>       <span class="co"># Split time [cite: 8]</span></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>n_pop9  <span class="ot">&lt;-</span> <span class="dv">1162</span>       <span class="co"># Ne for Pop 9 [cite: 10]</span></span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>n_pop1  <span class="ot">&lt;-</span> <span class="dv">13188</span>      <span class="co"># Ne for Pop 1 [cite: 11]</span></span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>n_anc   <span class="ot">&lt;-</span> <span class="dv">1895</span>       <span class="co"># Ne for Ancestral Pop [cite: 14]</span></span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>max_t   <span class="ot">&lt;-</span> <span class="dv">4000</span>       <span class="co"># Max time for Y-axis [cite: 6]</span></span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a><span class="do">## [cite_start]Migration rates [cite: 12, 13]</span></span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>mig_9_to_1 <span class="ot">&lt;-</span> <span class="st">"8.07e-05 / 1.064"</span></span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a>mig_1_to_9 <span class="ot">&lt;-</span> <span class="st">"5.33e-05 / 0.062"</span></span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a><span class="do">## Widened xlim to accommodate more space (0 to 40)</span></span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">1</span>), <span class="at">bg =</span> <span class="st">"white"</span>)</span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="cn">NULL</span>, <span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">40</span>), <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, max_t), </span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">""</span>, <span class="at">ylab =</span> <span class="st">"Time (generations)"</span>, </span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a>     <span class="at">xaxt =</span> <span class="st">"n"</span>, <span class="at">bty =</span> <span class="st">"n"</span>, <span class="at">las =</span> <span class="dv">1</span>)</span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-28"><a href="#cb25-28" aria-hidden="true" tabindex="-1"></a><span class="do">## --- 3. Draw Population Boxes ---</span></span>
<span id="cb25-29"><a href="#cb25-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-30"><a href="#cb25-30" aria-hidden="true" tabindex="-1"></a>scale_w <span class="ot">&lt;-</span> <span class="fl">0.0006</span> </span>
<span id="cb25-31"><a href="#cb25-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-32"><a href="#cb25-32" aria-hidden="true" tabindex="-1"></a><span class="do">## Pop 9 - Moved to x=8</span></span>
<span id="cb25-33"><a href="#cb25-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-34"><a href="#cb25-34" aria-hidden="true" tabindex="-1"></a>x9 <span class="ot">&lt;-</span> <span class="dv">8</span></span>
<span id="cb25-35"><a href="#cb25-35" aria-hidden="true" tabindex="-1"></a>w9 <span class="ot">&lt;-</span> n_pop9 <span class="sc">*</span> scale_w</span>
<span id="cb25-36"><a href="#cb25-36" aria-hidden="true" tabindex="-1"></a><span class="fu">rect</span>(x9 <span class="sc">-</span> w9, <span class="dv">0</span>, x9 <span class="sc">+</span> w9, t_div, <span class="at">col =</span> <span class="st">"#CAB2D6"</span>,</span>
<span id="cb25-37"><a href="#cb25-37" aria-hidden="true" tabindex="-1"></a>     <span class="at">border =</span> <span class="st">"#CAB2D6"</span>, <span class="at">lwd =</span> <span class="fl">1.5</span>)</span>
<span id="cb25-38"><a href="#cb25-38" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(x9, <span class="sc">-</span><span class="dv">200</span>, <span class="fu">expression</span>(Pop9<span class="sc">~</span>(N[e]<span class="sc">==</span><span class="dv">1162</span>)),</span>
<span id="cb25-39"><a href="#cb25-39" aria-hidden="true" tabindex="-1"></a>     <span class="at">xpd =</span> <span class="cn">TRUE</span>, <span class="at">font =</span> <span class="dv">2</span>)</span>
<span id="cb25-40"><a href="#cb25-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-41"><a href="#cb25-41" aria-hidden="true" tabindex="-1"></a><span class="do">## Pop 1 - Moved to x=32</span></span>
<span id="cb25-42"><a href="#cb25-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-43"><a href="#cb25-43" aria-hidden="true" tabindex="-1"></a>x1 <span class="ot">&lt;-</span> <span class="dv">32</span></span>
<span id="cb25-44"><a href="#cb25-44" aria-hidden="true" tabindex="-1"></a>w1 <span class="ot">&lt;-</span> n_pop1 <span class="sc">*</span> scale_w</span>
<span id="cb25-45"><a href="#cb25-45" aria-hidden="true" tabindex="-1"></a><span class="fu">rect</span>(x1 <span class="sc">-</span> w1, <span class="dv">0</span>, x1 <span class="sc">+</span> w1, t_div, <span class="at">col =</span> <span class="st">"#A6CEE3"</span>,</span>
<span id="cb25-46"><a href="#cb25-46" aria-hidden="true" tabindex="-1"></a>     <span class="at">border =</span> <span class="st">"#A6CEE3"</span>, <span class="at">lwd =</span> <span class="fl">1.5</span>)</span>
<span id="cb25-47"><a href="#cb25-47" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(x1, <span class="sc">-</span><span class="dv">200</span>, <span class="fu">expression</span>(Pop1<span class="sc">~</span>(N[e]<span class="sc">==</span><span class="dv">13188</span>)),</span>
<span id="cb25-48"><a href="#cb25-48" aria-hidden="true" tabindex="-1"></a>     <span class="at">xpd =</span> <span class="cn">TRUE</span>, <span class="at">font =</span> <span class="dv">2</span>)</span>
<span id="cb25-49"><a href="#cb25-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-50"><a href="#cb25-50" aria-hidden="true" tabindex="-1"></a><span class="do">## Ancestral Pop (Centered above Pop 1)</span></span>
<span id="cb25-51"><a href="#cb25-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-52"><a href="#cb25-52" aria-hidden="true" tabindex="-1"></a>wa <span class="ot">&lt;-</span> n_anc <span class="sc">*</span> scale_w</span>
<span id="cb25-53"><a href="#cb25-53" aria-hidden="true" tabindex="-1"></a><span class="fu">rect</span>(x1 <span class="sc">-</span> wa, t_div, x1 <span class="sc">+</span> wa, max_t, <span class="at">col =</span> <span class="st">"gray40"</span>,</span>
<span id="cb25-54"><a href="#cb25-54" aria-hidden="true" tabindex="-1"></a>     <span class="at">border =</span> <span class="st">"black"</span>)</span>
<span id="cb25-55"><a href="#cb25-55" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(x1, max_t <span class="sc">+</span> <span class="dv">150</span>, <span class="fu">expression</span>(Ancestral<span class="sc">~</span>size<span class="sc">~</span>(N[e]<span class="sc">==</span><span class="dv">1895</span>)),</span>
<span id="cb25-56"><a href="#cb25-56" aria-hidden="true" tabindex="-1"></a>     <span class="at">xpd =</span> <span class="cn">TRUE</span>, <span class="at">cex =</span> <span class="fl">0.9</span>)</span>
<span id="cb25-57"><a href="#cb25-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-58"><a href="#cb25-58" aria-hidden="true" tabindex="-1"></a><span class="do">## --- 4. Draw Connections &amp; Events ---</span></span>
<span id="cb25-59"><a href="#cb25-59" aria-hidden="true" tabindex="-1"></a><span class="do">## Split line adjusted to new coordinates</span></span>
<span id="cb25-60"><a href="#cb25-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-61"><a href="#cb25-61" aria-hidden="true" tabindex="-1"></a><span class="fu">segments</span>(x9, t_div, x1, t_div, <span class="at">lwd =</span> <span class="dv">2</span>, <span class="at">lty =</span> <span class="dv">1</span>)</span>
<span id="cb25-62"><a href="#cb25-62" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(x9, t_div, <span class="at">pch =</span> <span class="dv">21</span>, <span class="at">bg =</span> <span class="st">"black"</span>) </span>
<span id="cb25-63"><a href="#cb25-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-64"><a href="#cb25-64" aria-hidden="true" tabindex="-1"></a><span class="do">## T_DIV reference line</span></span>
<span id="cb25-65"><a href="#cb25-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-66"><a href="#cb25-66" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> t_div, <span class="at">lty =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="st">"red"</span>)</span>
<span id="cb25-67"><a href="#cb25-67" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(<span class="dv">0</span>, t_div <span class="sc">+</span> <span class="dv">140</span>, <span class="fu">expression</span>(<span class="fu">italic</span>(<span class="st">"Time of divergence"</span>)<span class="sc">==</span><span class="dv">3313</span>),</span>
<span id="cb25-68"><a href="#cb25-68" aria-hidden="true" tabindex="-1"></a>     <span class="at">col =</span> <span class="st">"red"</span>, <span class="at">adj =</span> <span class="dv">0</span>, <span class="at">cex =</span> <span class="fl">0.8</span>)</span>
<span id="cb25-69"><a href="#cb25-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-70"><a href="#cb25-70" aria-hidden="true" tabindex="-1"></a><span class="do">## --- 5. Add Migration Arrows (Gene Flow) ---</span></span>
<span id="cb25-71"><a href="#cb25-71" aria-hidden="true" tabindex="-1"></a><span class="do">## Middle point for labels</span></span>
<span id="cb25-72"><a href="#cb25-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-73"><a href="#cb25-73" aria-hidden="true" tabindex="-1"></a>mid_x <span class="ot">&lt;-</span> (x9 <span class="sc">+</span> x1) <span class="sc">/</span> <span class="dv">2</span></span>
<span id="cb25-74"><a href="#cb25-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-75"><a href="#cb25-75" aria-hidden="true" tabindex="-1"></a><span class="do">## Arrow from Pop 1 to Pop 9 (Leftward)</span></span>
<span id="cb25-76"><a href="#cb25-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-77"><a href="#cb25-77" aria-hidden="true" tabindex="-1"></a><span class="fu">arrows</span>(x1 <span class="sc">-</span> w1 <span class="sc">-</span> <span class="dv">1</span>, <span class="dv">1900</span>, x9 <span class="sc">+</span> w9 <span class="sc">+</span> <span class="dv">1</span>, <span class="dv">1900</span>, </span>
<span id="cb25-78"><a href="#cb25-78" aria-hidden="true" tabindex="-1"></a>       <span class="at">col =</span> <span class="st">"gold"</span>, <span class="at">lwd =</span> <span class="dv">1</span>, <span class="at">length =</span> <span class="fl">0.06</span>)</span>
<span id="cb25-79"><a href="#cb25-79" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(mid_x <span class="sc">-</span> <span class="dv">1</span>, <span class="dv">2010</span>, mig_1_to_9, <span class="at">col =</span> <span class="st">"goldenrod4"</span>, <span class="at">cex =</span> <span class="fl">0.7</span>)</span>
<span id="cb25-80"><a href="#cb25-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-81"><a href="#cb25-81" aria-hidden="true" tabindex="-1"></a><span class="do">## Arrow from Pop 9 to Pop 1 (Rightward)</span></span>
<span id="cb25-82"><a href="#cb25-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-83"><a href="#cb25-83" aria-hidden="true" tabindex="-1"></a><span class="fu">arrows</span>(x9 <span class="sc">+</span> w9 <span class="sc">+</span> <span class="dv">1</span>, <span class="dv">1800</span>, x1 <span class="sc">-</span> w1 <span class="sc">-</span> <span class="dv">1</span>, <span class="dv">1800</span>, </span>
<span id="cb25-84"><a href="#cb25-84" aria-hidden="true" tabindex="-1"></a>       <span class="at">col =</span> <span class="st">"orange"</span>, <span class="at">lwd =</span> <span class="dv">1</span>, <span class="at">length =</span> <span class="fl">0.06</span>)</span>
<span id="cb25-85"><a href="#cb25-85" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(x9 <span class="sc">+</span> w9 <span class="sc">+</span> <span class="dv">5</span>, <span class="dv">1710</span>, mig_9_to_1, <span class="at">col =</span> <span class="st">"darkorange3"</span>, <span class="at">cex =</span> <span class="fl">0.7</span>)</span>
<span id="cb25-86"><a href="#cb25-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-87"><a href="#cb25-87" aria-hidden="true" tabindex="-1"></a><span class="fu">mtext</span>(<span class="st">"A"</span>, <span class="at">side =</span> <span class="dv">2</span>, <span class="at">at =</span> <span class="dv">5000</span>, <span class="at">line =</span> <span class="fl">3.2</span>, <span class="at">las =</span> <span class="dv">1</span>)</span>
<span id="cb25-88"><a href="#cb25-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-89"><a href="#cb25-89" aria-hidden="true" tabindex="-1"></a><span class="do">## 2D-SFS ##</span></span>
<span id="cb25-90"><a href="#cb25-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-91"><a href="#cb25-91" aria-hidden="true" tabindex="-1"></a><span class="do">## 1. Configuration: Set your file names and labels here</span></span>
<span id="cb25-92"><a href="#cb25-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-93"><a href="#cb25-93" aria-hidden="true" tabindex="-1"></a>infile_path <span class="ot">&lt;-</span> <span class="st">"recent_geneflow_jointMAFpop1_0.obs"</span></span>
<span id="cb25-94"><a href="#cb25-94" aria-hidden="true" tabindex="-1"></a>output_name <span class="ot">&lt;-</span> <span class="st">"my_custom_plot.png"</span>                </span>
<span id="cb25-95"><a href="#cb25-95" aria-hidden="true" tabindex="-1"></a>x_label <span class="ot">&lt;-</span> <span class="st">"Pop1"</span>                                  </span>
<span id="cb25-96"><a href="#cb25-96" aria-hidden="true" tabindex="-1"></a>y_label <span class="ot">&lt;-</span> <span class="st">"Pop9"</span>                                  </span>
<span id="cb25-97"><a href="#cb25-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-98"><a href="#cb25-98" aria-hidden="true" tabindex="-1"></a><span class="do">## 2. Define Color Palettes (Extracted from SFStools.R) </span></span>
<span id="cb25-99"><a href="#cb25-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-100"><a href="#cb25-100" aria-hidden="true" tabindex="-1"></a>ramp99 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#A50026"</span>,<span class="st">"#AA0426"</span>,<span class="st">"#AF0926"</span>,</span>
<span id="cb25-101"><a href="#cb25-101" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#B40E26"</span>,<span class="st">"#B91326"</span>,<span class="st">"#BE1826"</span>,</span>
<span id="cb25-102"><a href="#cb25-102" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#C31D26"</span>,<span class="st">"#C82226"</span>,<span class="st">"#CD2726"</span>,</span>
<span id="cb25-103"><a href="#cb25-103" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#D22C26"</span>,<span class="st">"#D73127"</span>,<span class="st">"#DA372A"</span>,</span>
<span id="cb25-104"><a href="#cb25-104" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#DD3D2D"</span>,<span class="st">"#E04330"</span>,<span class="st">"#E34A33"</span>,</span>
<span id="cb25-105"><a href="#cb25-105" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#E65035"</span>,<span class="st">"#E95638"</span>,<span class="st">"#EC5C3B"</span>,</span>
<span id="cb25-106"><a href="#cb25-106" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#EF633E"</span>,<span class="st">"#F26941"</span>,<span class="st">"#F46F44"</span>,</span>
<span id="cb25-107"><a href="#cb25-107" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#F57647"</span>,<span class="st">"#F67C4A"</span>,<span class="st">"#F7834D"</span>,</span>
<span id="cb25-108"><a href="#cb25-108" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#F88A50"</span>,<span class="st">"#F89053"</span>,<span class="st">"#F99756"</span>,</span>
<span id="cb25-109"><a href="#cb25-109" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#FA9E59"</span>,<span class="st">"#FBA45C"</span>,<span class="st">"#FCAB5F"</span>,</span>
<span id="cb25-110"><a href="#cb25-110" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#FDB163"</span>,<span class="st">"#FDB668"</span>,<span class="st">"#FDBB6D"</span>,</span>
<span id="cb25-111"><a href="#cb25-111" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#FDC072"</span>,<span class="st">"#FDC577"</span>,<span class="st">"#FDCA7B"</span>,</span>
<span id="cb25-112"><a href="#cb25-112" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#FDCF80"</span>,<span class="st">"#FDD485"</span>,<span class="st">"#FDD98A"</span>,</span>
<span id="cb25-113"><a href="#cb25-113" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#FDDE8F"</span>,<span class="st">"#FEE293"</span>,<span class="st">"#FEE598"</span>,</span>
<span id="cb25-114"><a href="#cb25-114" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#FEE89D"</span>,<span class="st">"#FEECA2"</span>,<span class="st">"#FEEFA7"</span>,</span>
<span id="cb25-115"><a href="#cb25-115" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#FEF2AB"</span>,<span class="st">"#FEF5B0"</span>,<span class="st">"#FEF8B5"</span>,</span>
<span id="cb25-116"><a href="#cb25-116" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#FEFBBA"</span>,<span class="st">"#FFFEBE"</span>,<span class="st">"#FBFDC4"</span>,</span>
<span id="cb25-117"><a href="#cb25-117" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#F8FCCA"</span>,<span class="st">"#F5FBD0"</span>,<span class="st">"#F2FAD6"</span>,</span>
<span id="cb25-118"><a href="#cb25-118" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#EFF8DC"</span>,<span class="st">"#ECF7E1"</span>,<span class="st">"#E8F6E7"</span>,</span>
<span id="cb25-119"><a href="#cb25-119" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#E5F5ED"</span>,<span class="st">"#E2F3F3"</span>,<span class="st">"#DEF2F7"</span>,</span>
<span id="cb25-120"><a href="#cb25-120" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#D9EFF6"</span>,<span class="st">"#D4EDF4"</span>,<span class="st">"#CEEAF3"</span>,</span>
<span id="cb25-121"><a href="#cb25-121" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#C9E7F1"</span>,<span class="st">"#C3E5F0"</span>,<span class="st">"#BEE2EE"</span>,</span>
<span id="cb25-122"><a href="#cb25-122" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#B9DFEC"</span>,<span class="st">"#B3DDEB"</span>,<span class="st">"#AEDAE9"</span>,</span>
<span id="cb25-123"><a href="#cb25-123" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#A8D7E8"</span>,<span class="st">"#A3D2E5"</span>,<span class="st">"#9DCEE3"</span>,</span>
<span id="cb25-124"><a href="#cb25-124" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#97C9E0"</span>,<span class="st">"#92C5DE"</span>,<span class="st">"#8CC0DB"</span>,</span>
<span id="cb25-125"><a href="#cb25-125" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#87BCD9"</span>,<span class="st">"#81B7D6"</span>,<span class="st">"#7BB3D4"</span>,</span>
<span id="cb25-126"><a href="#cb25-126" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#76AED1"</span>,<span class="st">"#71A9CF"</span>,<span class="st">"#6CA3CC"</span>,</span>
<span id="cb25-127"><a href="#cb25-127" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#679EC9"</span>,<span class="st">"#6298C6"</span>,<span class="st">"#5D92C3"</span>,</span>
<span id="cb25-128"><a href="#cb25-128" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#598DC0"</span>,<span class="st">"#5487BD"</span>,<span class="st">"#4F81BA"</span>,</span>
<span id="cb25-129"><a href="#cb25-129" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#4A7BB7"</span>,<span class="st">"#4576B4"</span>,<span class="st">"#436FB1"</span>,</span>
<span id="cb25-130"><a href="#cb25-130" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#4169AE"</span>,<span class="st">"#3F63AB"</span>,<span class="st">"#3D5CA7"</span>,</span>
<span id="cb25-131"><a href="#cb25-131" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#3B56A4"</span>,<span class="st">"#394FA1"</span>,<span class="st">"#37499E"</span>,</span>
<span id="cb25-132"><a href="#cb25-132" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#35429B"</span>,<span class="st">"#333C98"</span>,<span class="st">"#313695"</span>)</span>
<span id="cb25-133"><a href="#cb25-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-134"><a href="#cb25-134" aria-hidden="true" tabindex="-1"></a>ramp30 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#A50026"</span>,<span class="st">"#B61026"</span>,<span class="st">"#C72126"</span>,</span>
<span id="cb25-135"><a href="#cb25-135" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#D83227"</span>,<span class="st">"#E24731"</span>,<span class="st">"#EC5C3B"</span>,</span>
<span id="cb25-136"><a href="#cb25-136" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#F47145"</span>,<span class="st">"#F7874F"</span>,<span class="st">"#FA9E59"</span>,</span>
<span id="cb25-137"><a href="#cb25-137" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#FDB365"</span>,<span class="st">"#FDC476"</span>,<span class="st">"#FDD586"</span>,</span>
<span id="cb25-138"><a href="#cb25-138" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#FEE496"</span>,<span class="st">"#FEEEA6"</span>,<span class="st">"#FEF9B6"</span>,</span>
<span id="cb25-139"><a href="#cb25-139" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#F9FCC8"</span>,<span class="st">"#EEF8DC"</span>,<span class="st">"#E4F4F0"</span>,</span>
<span id="cb25-140"><a href="#cb25-140" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#D5EDF4"</span>,<span class="st">"#C2E4EF"</span>,<span class="st">"#B0DBEA"</span>,</span>
<span id="cb25-141"><a href="#cb25-141" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#9DCEE3"</span>,<span class="st">"#8ABFDA"</span>,<span class="st">"#77B0D2"</span>,</span>
<span id="cb25-142"><a href="#cb25-142" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#679DC9"</span>,<span class="st">"#568ABF"</span>,<span class="st">"#4676B5"</span>,</span>
<span id="cb25-143"><a href="#cb25-143" aria-hidden="true" tabindex="-1"></a>            <span class="st">"#3E61AA"</span>,<span class="st">"#374B9F"</span>,<span class="st">"#313695"</span>)</span>
<span id="cb25-144"><a href="#cb25-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-145"><a href="#cb25-145" aria-hidden="true" tabindex="-1"></a><span class="do">## 3. Load Data </span></span>
<span id="cb25-146"><a href="#cb25-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-147"><a href="#cb25-147" aria-hidden="true" tabindex="-1"></a>obs <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">read.table</span>(infile_path, <span class="at">skip =</span> <span class="dv">1</span>, <span class="at">header =</span> <span class="cn">TRUE</span>))</span>
<span id="cb25-148"><a href="#cb25-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-149"><a href="#cb25-149" aria-hidden="true" tabindex="-1"></a><span class="do">## 4. Plotting Setup </span></span>
<span id="cb25-150"><a href="#cb25-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-151"><a href="#cb25-151" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="fl">5.1</span>, <span class="fl">4.1</span>, <span class="fl">4.1</span>, <span class="dv">4</span>)) <span class="co"># Standard margins for centering</span></span>
<span id="cb25-152"><a href="#cb25-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-153"><a href="#cb25-153" aria-hidden="true" tabindex="-1"></a><span class="do">## Calculate log scale for colors </span></span>
<span id="cb25-154"><a href="#cb25-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-155"><a href="#cb25-155" aria-hidden="true" tabindex="-1"></a>uplog <span class="ot">&lt;-</span> <span class="fu">log10</span>(<span class="fu">max</span>(obs, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">+</span> <span class="fl">0.00001</span>)</span>
<span id="cb25-156"><a href="#cb25-156" aria-hidden="true" tabindex="-1"></a>br <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">10</span><span class="sc">^</span>(<span class="fu">seq</span>(<span class="fu">log10</span>(<span class="fl">0.5</span>), uplog, <span class="at">length.out =</span> <span class="dv">100</span>)))</span>
<span id="cb25-157"><a href="#cb25-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-158"><a href="#cb25-158" aria-hidden="true" tabindex="-1"></a><span class="do">## 5. Generate Heatmap </span></span>
<span id="cb25-159"><a href="#cb25-159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-160"><a href="#cb25-160" aria-hidden="true" tabindex="-1"></a><span class="fu">image</span>(<span class="dv">0</span><span class="sc">:</span>(<span class="fu">dim</span>(obs)[<span class="dv">1</span>]<span class="sc">-</span><span class="dv">1</span>), <span class="dv">0</span><span class="sc">:</span>(<span class="fu">dim</span>(obs)[<span class="dv">2</span>]<span class="sc">-</span><span class="dv">1</span>), obs, <span class="at">col =</span> <span class="fu">rev</span>(ramp99), </span>
<span id="cb25-161"><a href="#cb25-161" aria-hidden="true" tabindex="-1"></a>      <span class="at">zlim =</span> <span class="fl">0.5</span>, <span class="at">breaks =</span> br, <span class="at">axes =</span> <span class="cn">FALSE</span>, <span class="at">xlab =</span> <span class="st">""</span>, <span class="at">ylab =</span> <span class="st">""</span>, <span class="at">main =</span> <span class="st">""</span>)</span>
<span id="cb25-162"><a href="#cb25-162" aria-hidden="true" tabindex="-1"></a><span class="fu">box</span>()</span>
<span id="cb25-163"><a href="#cb25-163" aria-hidden="true" tabindex="-1"></a><span class="fu">axis</span>(<span class="dv">2</span>, <span class="at">las =</span> <span class="dv">1</span>)</span>
<span id="cb25-164"><a href="#cb25-164" aria-hidden="true" tabindex="-1"></a><span class="fu">axis</span>(<span class="dv">1</span>)</span>
<span id="cb25-165"><a href="#cb25-165" aria-hidden="true" tabindex="-1"></a><span class="fu">mtext</span>(y_label, <span class="dv">2</span>, <span class="at">line =</span> <span class="fl">2.5</span>, <span class="at">font =</span> <span class="dv">1</span>, <span class="at">cex =</span> <span class="fl">0.8</span>)</span>
<span id="cb25-166"><a href="#cb25-166" aria-hidden="true" tabindex="-1"></a><span class="fu">mtext</span>(x_label, <span class="dv">1</span>, <span class="at">line =</span> <span class="fl">2.5</span>, <span class="at">font =</span> <span class="dv">1</span>, <span class="at">cex =</span> <span class="fl">0.8</span>)</span>
<span id="cb25-167"><a href="#cb25-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-168"><a href="#cb25-168" aria-hidden="true" tabindex="-1"></a><span class="do">## 6. Add the Legend Scale </span></span>
<span id="cb25-169"><a href="#cb25-169" aria-hidden="true" tabindex="-1"></a>logax <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">5</span>) <span class="sc">*</span> <span class="fu">rep</span>(<span class="dv">10</span><span class="sc">^</span>(<span class="fu">seq</span>(<span class="dv">0</span>, <span class="fu">floor</span>(uplog))), <span class="at">each =</span> <span class="dv">3</span>)</span>
<span id="cb25-170"><a href="#cb25-170" aria-hidden="true" tabindex="-1"></a>logax <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.5</span>, logax[logax <span class="sc">&lt;</span> <span class="dv">10</span><span class="sc">^</span>(uplog)], <span class="fu">round</span>(<span class="dv">10</span><span class="sc">^</span>(uplog), <span class="dv">0</span>))</span>
<span id="cb25-171"><a href="#cb25-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-172"><a href="#cb25-172" aria-hidden="true" tabindex="-1"></a><span class="fu">axis</span>(<span class="dv">4</span>, <span class="at">at =</span> <span class="fu">grconvertY</span>(<span class="dv">0</span>, <span class="st">"npc"</span>, <span class="st">"user"</span>) <span class="sc">+</span> <span class="fu">diff</span>(<span class="fu">grconvertY</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">1</span>, <span class="st">"npc"</span>, <span class="st">"user"</span>)) <span class="sc">*</span></span>
<span id="cb25-173"><a href="#cb25-173" aria-hidden="true" tabindex="-1"></a>       (<span class="fu">log10</span>(logax) <span class="sc">-</span> <span class="fu">log10</span>(logax)[<span class="dv">1</span>]) <span class="sc">/</span> <span class="fu">diff</span>(<span class="fu">log10</span>(logax[<span class="fu">c</span>(<span class="dv">1</span>, <span class="fu">length</span>(logax))])),</span>
<span id="cb25-174"><a href="#cb25-174" aria-hidden="true" tabindex="-1"></a>     <span class="fu">round</span>(logax, <span class="dv">1</span>), <span class="at">las =</span> <span class="dv">2</span>, <span class="at">line =</span> <span class="fl">1.2</span>, <span class="at">cex.axis =</span> <span class="fl">0.6</span>)</span>
<span id="cb25-175"><a href="#cb25-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-176"><a href="#cb25-176" aria-hidden="true" tabindex="-1"></a><span class="fu">rasterImage</span>(<span class="fu">as.raster</span>(<span class="fu">matrix</span>(ramp30, <span class="at">ncol =</span> <span class="dv">1</span>)),</span>
<span id="cb25-177"><a href="#cb25-177" aria-hidden="true" tabindex="-1"></a>            <span class="fu">par</span>(<span class="st">'usr'</span>)[<span class="dv">2</span>] <span class="sc">+</span> <span class="fl">0.6</span> <span class="sc">*</span> <span class="fu">diff</span>(<span class="fu">grconvertX</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">1</span>, <span class="st">'inches'</span>, <span class="st">'user'</span>)) <span class="sc">*</span> <span class="fu">par</span>(<span class="st">'cin'</span>)[<span class="dv">2</span>],</span>
<span id="cb25-178"><a href="#cb25-178" aria-hidden="true" tabindex="-1"></a>            <span class="fu">grconvertY</span>(<span class="dv">0</span>, <span class="st">"npc"</span>, <span class="st">"user"</span>),</span>
<span id="cb25-179"><a href="#cb25-179" aria-hidden="true" tabindex="-1"></a>            <span class="fu">par</span>(<span class="st">'usr'</span>)[<span class="dv">2</span>] <span class="sc">+</span> <span class="fl">1.2</span> <span class="sc">*</span> <span class="fu">diff</span>(<span class="fu">grconvertX</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">1</span>, <span class="st">'inches'</span>, <span class="st">'user'</span>)) <span class="sc">*</span> <span class="fu">par</span>(<span class="st">'cin'</span>)[<span class="dv">2</span>],</span>
<span id="cb25-180"><a href="#cb25-180" aria-hidden="true" tabindex="-1"></a>            <span class="fu">grconvertY</span>(<span class="dv">1</span>, <span class="st">"npc"</span>, <span class="st">"user"</span>), <span class="at">xpd =</span> <span class="cn">NA</span>)</span>
<span id="cb25-181"><a href="#cb25-181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-182"><a href="#cb25-182" aria-hidden="true" tabindex="-1"></a><span class="fu">mtext</span>(<span class="fu">expression</span>(N[loci]), <span class="dv">3</span>,</span>
<span id="cb25-183"><a href="#cb25-183" aria-hidden="true" tabindex="-1"></a>      <span class="at">at =</span> <span class="fu">par</span>(<span class="st">'usr'</span>)[<span class="dv">2</span>] <span class="sc">+</span> <span class="fl">0.8</span> <span class="sc">*</span> <span class="fu">diff</span>(<span class="fu">grconvertX</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">1</span>, <span class="st">'inches'</span>, <span class="st">'user'</span>)) <span class="sc">*</span> <span class="fu">par</span>(<span class="st">'cin'</span>)[<span class="dv">2</span>], </span>
<span id="cb25-184"><a href="#cb25-184" aria-hidden="true" tabindex="-1"></a>      <span class="at">adj =</span> <span class="dv">0</span>, <span class="at">line =</span> <span class="fl">0.5</span>, <span class="at">cex =</span> <span class="fl">0.7</span>)</span>
<span id="cb25-185"><a href="#cb25-185" aria-hidden="true" tabindex="-1"></a><span class="fu">mtext</span>(<span class="st">"B"</span>, <span class="at">side =</span> <span class="dv">2</span>, <span class="at">at =</span> <span class="dv">60</span>, <span class="at">line =</span> <span class="fl">3.2</span>, <span class="at">las =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><img src="imgs/recent_geneflow_model_2DSFS.png" class="img-fluid"></p>
<p><strong>This script was designed to produce a genind object per population of aRanSyl required for the estimates of <span class="math inline">\(N_e\)</span> described below</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="do">## loading project</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"/gpfs/gibbs/project/skelly/dp996/Wood-frog_ddRADseq/data/aRanSyl_genind.RData"</span>)</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"/gpfs/gibbs/project/skelly/dp996/Wood-frog_ddRADseq/data/aRanSyl_metadata.csv"</span>)</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a><span class="fu">tapply</span>(meta<span class="sc">$</span>seq.run, meta<span class="sc">$</span>pop, length)</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>meta<span class="sc">$</span>pop <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(meta<span class="sc">$</span>pop)</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>aRanSyl_genind_list <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">"list"</span>, <span class="fu">length</span>(<span class="fu">levels</span>(meta<span class="sc">$</span>pop)))</span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>idx <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">levels</span>(meta<span class="sc">$</span>pop)){</span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>aRanSyl_genind_pop <span class="ot">&lt;-</span> aRanSyl_genind</span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a>pop <span class="ot">&lt;-</span> meta[meta<span class="sc">$</span>pop <span class="sc">==</span> i, ]</span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a>pop.ind <span class="ot">&lt;-</span> pop<span class="sc">$</span>id</span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a>filter.ind <span class="ot">&lt;-</span> <span class="fu">indNames</span>(aRanSyl_genind_pop)[<span class="fu">indNames</span>(aRanSyl_genind_pop) <span class="sc">%in%</span> pop.ind]</span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a>aRanSyl_genind_pop <span class="ot">&lt;-</span> aRanSyl_genind_pop[filter.ind, ]</span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a>aRanSyl_genind_pop<span class="sc">$</span>pop <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="fu">rep</span>(i, <span class="fu">length</span>(pop.ind)))</span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a>aRanSyl_genind_list[[idx]] <span class="ot">&lt;-</span> aRanSyl_genind_pop</span>
<span id="cb26-23"><a href="#cb26-23" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(aRanSyl_genind_list)[[idx]] <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">'aRanSyl_genind_pop'</span>,<span class="st">'_'</span>,i, <span class="at">sep =</span> <span class="st">''</span>)</span>
<span id="cb26-24"><a href="#cb26-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-25"><a href="#cb26-25" aria-hidden="true" tabindex="-1"></a>idx <span class="ot">=</span> idx <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb26-26"><a href="#cb26-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-27"><a href="#cb26-27" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb26-28"><a href="#cb26-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-29"><a href="#cb26-29" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(aRanSyl_genind_list, <span class="at">file =</span> <span class="st">"/gpfs/gibbs/project/skelly/dp996/Wood-frog_ddRADseq/data/aRanSyl_genind_list.RData"</span>)</span>
<span id="cb26-30"><a href="#cb26-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-31"><a href="#cb26-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-32"><a href="#cb26-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-33"><a href="#cb26-33" aria-hidden="true" tabindex="-1"></a><span class="do">## converting to genlight objects</span></span>
<span id="cb26-34"><a href="#cb26-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-35"><a href="#cb26-35" aria-hidden="true" tabindex="-1"></a>aRanSyl_gl_list <span class="ot">&lt;-</span> aRanSyl_genind_list</span>
<span id="cb26-36"><a href="#cb26-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-37"><a href="#cb26-37" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">levels</span>(meta<span class="sc">$</span>pop))){</span>
<span id="cb26-38"><a href="#cb26-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-39"><a href="#cb26-39" aria-hidden="true" tabindex="-1"></a>aRanSyl_gl_list[[i]] <span class="ot">&lt;-</span> <span class="fu">gi2gl</span>(aRanSyl_gl_list[[i]], <span class="at">parallel =</span> <span class="cn">FALSE</span>, <span class="at">verbose =</span> <span class="cn">NULL</span>)</span>
<span id="cb26-40"><a href="#cb26-40" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(aRanSyl_gl_list)[[i]] <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">'aRanSyl_gl_pop'</span>,<span class="st">'_'</span>,i, <span class="at">sep =</span> <span class="st">''</span>)</span>
<span id="cb26-41"><a href="#cb26-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-42"><a href="#cb26-42" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb26-43"><a href="#cb26-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-44"><a href="#cb26-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-45"><a href="#cb26-45" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(aRanSyl_gl_list, <span class="at">file =</span> <span class="st">"/gpfs/gibbs/project/skelly/dp996/Wood-frog_ddRADseq/data/aRanSyl_gl_list.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>This script was designed to estimate <span class="math inline">\(N_e\)</span> across populations of aRanSyl</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="do">## set paths for different versions</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>path.binaries <span class="ot">&lt;-</span> <span class="st">"/Users/dpadil10/ASU Dropbox/Dylan Padilla/Yale/aRanSyl-PopGen/binaries/"</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"/Users/dpadil10/ASU Dropbox/Dylan Padilla/Yale/aRanSyl-PopGen/data/aRanSyl_gl_list.RData"</span>)</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>aRanSyl_gl_list</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>aRanSyl_Ne_list <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">"list"</span>, <span class="dv">9</span>)</span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>){</span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>    <span class="do">## current effective population size</span></span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>    aRanSyl_gl <span class="ot">&lt;-</span> <span class="fu">gl.filter.allna</span>(aRanSyl_gl_list[[i]])</span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a>    aRanSyl_gl</span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a>    aRanSyl_Ne <span class="ot">&lt;-</span> <span class="fu">gl.LDNe</span>(aRanSyl_gl,  <span class="at">outfile =</span> <span class="st">"popsLD.txt"</span>,</span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a>                      <span class="at">neest.path =</span> path.binaries,</span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a>                      <span class="at">critical =</span> <span class="fu">c</span>(<span class="fl">0.01</span>, <span class="fl">0.02</span>, <span class="fl">0.05</span>),</span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a>                      <span class="at">singleton.rm =</span> <span class="cn">TRUE</span>, <span class="at">mating =</span> <span class="st">"random"</span>, <span class="at">plot.out =</span> <span class="cn">FALSE</span>)</span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a>    aRanSyl_Ne_list[[i]] <span class="ot">&lt;-</span> aRanSyl_Ne</span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(aRanSyl_Ne_list)[[i]] <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"aRanSyl_Ne_pop"</span>,<span class="st">"_"</span>,i, <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb27-22"><a href="#cb27-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">save</span>(aRanSyl_Ne_list, <span class="at">file =</span> <span class="st">"/Users/dpadil10/ASU Dropbox/Dylan Padilla/Yale/aRanSyl-PopGen/outputs/aRanSyl_Ne/aRanSyl_Ne_list.RData"</span>)</span>
<span id="cb27-23"><a href="#cb27-23" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="do">## loading project</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"/Users/dpadil10/ASU Dropbox/Dylan Padilla/Yale/aRanSyl-PopGen/data/aRanSyl_genind_pops.RData"</span>)</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"/Users/dpadil10/ASU Dropbox/Dylan Padilla/Yale/aRanSyl-PopGen/data/aRanSyl_metadata.csv"</span>)</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"/Users/dpadil10/ASU Dropbox/Dylan Padilla/Yale/aRanSyl-PopGen/outputs/aRanSyl_Ne/aRanSyl_Ne_list.RData"</span>)</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>barplot.data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="st">"Ne_0.05"</span> <span class="ot">=</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(aRanSyl_Ne_list)),</span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"Ne_0.02"</span> <span class="ot">=</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(aRanSyl_Ne_list)),</span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"Ne_0.01"</span> <span class="ot">=</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(aRanSyl_Ne_list)),</span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"Pops"</span> <span class="ot">=</span> <span class="fu">paste</span>(<span class="fu">rep</span>(<span class="st">"Pop"</span>, <span class="dv">9</span>),<span class="fu">seq</span>(<span class="dv">1</span>, <span class="dv">9</span>, <span class="dv">1</span>), <span class="at">sep =</span> <span class="st">""</span>))</span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>barplot.data</span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a><span class="do">## Getting the Ne values</span></span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(aRanSyl_Ne_list)){</span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>    obj <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(aRanSyl_Ne_list[[i]])</span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a>    barplot.data[i, <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>)] <span class="ot">&lt;-</span>  <span class="fu">as.numeric</span>(obj[<span class="dv">6</span>, <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">4</span>)])</span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(barplot.data)</span>
<span id="cb28-22"><a href="#cb28-22" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(barplot.data) <span class="ot">&lt;-</span> barplot.data[ , <span class="dv">4</span>]</span>
<span id="cb28-23"><a href="#cb28-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-24"><a href="#cb28-24" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(meta)</span>
<span id="cb28-25"><a href="#cb28-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-26"><a href="#cb28-26" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(meta)[<span class="dv">17</span>] <span class="ot">&lt;-</span> <span class="st">"Pops"</span></span>
<span id="cb28-27"><a href="#cb28-27" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(meta)</span>
<span id="cb28-28"><a href="#cb28-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-29"><a href="#cb28-29" aria-hidden="true" tabindex="-1"></a>meta<span class="sc">$</span>Pops <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(meta<span class="sc">$</span>Pops)</span>
<span id="cb28-30"><a href="#cb28-30" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(meta<span class="sc">$</span>Pops) <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">rep</span>(<span class="st">"Pop"</span>, <span class="dv">9</span>), <span class="fu">seq</span>(<span class="dv">1</span>, <span class="dv">9</span>, <span class="dv">1</span>), <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb28-31"><a href="#cb28-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-32"><a href="#cb28-32" aria-hidden="true" tabindex="-1"></a><span class="fu">tapply</span>(meta<span class="sc">$</span>id, meta<span class="sc">$</span>Pops, length)</span>
<span id="cb28-33"><a href="#cb28-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-34"><a href="#cb28-34" aria-hidden="true" tabindex="-1"></a>submeta <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="st">"longitude"</span> <span class="ot">=</span> meta<span class="sc">$</span>longitude, <span class="st">"latitude"</span> <span class="ot">=</span> meta<span class="sc">$</span>latitude,</span>
<span id="cb28-35"><a href="#cb28-35" aria-hidden="true" tabindex="-1"></a>                      <span class="st">"Pops"</span> <span class="ot">=</span> meta<span class="sc">$</span>Pops)</span>
<span id="cb28-36"><a href="#cb28-36" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">merge</span>(submeta, barplot.data, <span class="at">by =</span> <span class="st">"Pops"</span>)</span>
<span id="cb28-37"><a href="#cb28-37" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(data)</span>
<span id="cb28-38"><a href="#cb28-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-39"><a href="#cb28-39" aria-hidden="true" tabindex="-1"></a>barplot.data <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(barplot.data[ , <span class="sc">-</span><span class="dv">4</span>])</span>
<span id="cb28-40"><a href="#cb28-40" aria-hidden="true" tabindex="-1"></a>barplot.data</span>
<span id="cb28-41"><a href="#cb28-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-42"><a href="#cb28-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-43"><a href="#cb28-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-44"><a href="#cb28-44" aria-hidden="true" tabindex="-1"></a><span class="do">## getting the CIs</span></span>
<span id="cb28-45"><a href="#cb28-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-46"><a href="#cb28-46" aria-hidden="true" tabindex="-1"></a>ci.data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="st">"CI_lower_0.05"</span> <span class="ot">=</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(aRanSyl_Ne_list)),</span>
<span id="cb28-47"><a href="#cb28-47" aria-hidden="true" tabindex="-1"></a>                      <span class="st">"CI_upper_0.05"</span> <span class="ot">=</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(aRanSyl_Ne_list)),</span>
<span id="cb28-48"><a href="#cb28-48" aria-hidden="true" tabindex="-1"></a>                      <span class="st">"CI_lower_0.02"</span> <span class="ot">=</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(aRanSyl_Ne_list)),</span>
<span id="cb28-49"><a href="#cb28-49" aria-hidden="true" tabindex="-1"></a>                      <span class="st">"CI_upper_0.02"</span> <span class="ot">=</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(aRanSyl_Ne_list)),</span>
<span id="cb28-50"><a href="#cb28-50" aria-hidden="true" tabindex="-1"></a>                      <span class="st">"CI_lower_0.01"</span> <span class="ot">=</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(aRanSyl_Ne_list)),</span>
<span id="cb28-51"><a href="#cb28-51" aria-hidden="true" tabindex="-1"></a>                      <span class="st">"CI_upper_0.01"</span> <span class="ot">=</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(aRanSyl_Ne_list)),</span>
<span id="cb28-52"><a href="#cb28-52" aria-hidden="true" tabindex="-1"></a>                      <span class="st">"Pops"</span> <span class="ot">=</span> <span class="fu">paste</span>(<span class="fu">rep</span>(<span class="st">"Pop"</span>, <span class="dv">9</span>),<span class="fu">seq</span>(<span class="dv">1</span>, <span class="dv">9</span>, <span class="dv">1</span>), <span class="at">sep =</span> <span class="st">""</span>))</span>
<span id="cb28-53"><a href="#cb28-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-54"><a href="#cb28-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-55"><a href="#cb28-55" aria-hidden="true" tabindex="-1"></a>ci.data</span>
<span id="cb28-56"><a href="#cb28-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-57"><a href="#cb28-57" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(aRanSyl_Ne_list)){</span>
<span id="cb28-58"><a href="#cb28-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-59"><a href="#cb28-59" aria-hidden="true" tabindex="-1"></a>    obj <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(aRanSyl_Ne_list[[i]])</span>
<span id="cb28-60"><a href="#cb28-60" aria-hidden="true" tabindex="-1"></a>    ci.data[i, <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>] <span class="ot">&lt;-</span>  <span class="fu">c</span>(<span class="fu">as.numeric</span>(obj[<span class="dv">7</span><span class="sc">:</span><span class="dv">8</span>, <span class="dv">2</span>]),</span>
<span id="cb28-61"><a href="#cb28-61" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">as.numeric</span>(obj[<span class="dv">7</span><span class="sc">:</span><span class="dv">8</span>, <span class="dv">3</span>]),</span>
<span id="cb28-62"><a href="#cb28-62" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">as.numeric</span>(obj[<span class="dv">7</span><span class="sc">:</span><span class="dv">8</span>, <span class="dv">4</span>]))</span>
<span id="cb28-63"><a href="#cb28-63" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb28-64"><a href="#cb28-64" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(ci.data)</span>
<span id="cb28-65"><a href="#cb28-65" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(ci.data) <span class="ot">&lt;-</span> ci.data[ , <span class="dv">7</span>]</span>
<span id="cb28-66"><a href="#cb28-66" aria-hidden="true" tabindex="-1"></a>ci.data</span>
<span id="cb28-67"><a href="#cb28-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-68"><a href="#cb28-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-69"><a href="#cb28-69" aria-hidden="true" tabindex="-1"></a><span class="do">## extracting historical data you can do this using the function</span></span>
<span id="cb28-70"><a href="#cb28-70" aria-hidden="true" tabindex="-1"></a><span class="do">## cmip6_world in the geodata package or you can manually do on</span></span>
<span id="cb28-71"><a href="#cb28-71" aria-hidden="true" tabindex="-1"></a><span class="do">##the WorldClim website.  In my case, I downloaded all the 19 bio</span></span>
<span id="cb28-72"><a href="#cb28-72" aria-hidden="true" tabindex="-1"></a><span class="do">## variables from the website, but I used the geodata to download</span></span>
<span id="cb28-73"><a href="#cb28-73" aria-hidden="true" tabindex="-1"></a><span class="do">##the future climate data</span></span>
<span id="cb28-74"><a href="#cb28-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-75"><a href="#cb28-75" aria-hidden="true" tabindex="-1"></a><span class="do">## download global bioclimatic data from worldclim </span></span>
<span id="cb28-76"><a href="#cb28-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-77"><a href="#cb28-77" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"/Users/dpadil10/ASU Dropbox/Dylan Padilla/Yale/aRanSyl-PopGen/data/climate/wc2.1_30s"</span>,</span>
<span id="cb28-78"><a href="#cb28-78" aria-hidden="true" tabindex="-1"></a>                    <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb28-79"><a href="#cb28-79" aria-hidden="true" tabindex="-1"></a>files</span>
<span id="cb28-80"><a href="#cb28-80" aria-hidden="true" tabindex="-1"></a>climate.var <span class="ot">&lt;-</span> <span class="fu">stack</span>(files)</span>
<span id="cb28-81"><a href="#cb28-81" aria-hidden="true" tabindex="-1"></a>climate.var</span>
<span id="cb28-82"><a href="#cb28-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-83"><a href="#cb28-83" aria-hidden="true" tabindex="-1"></a>file <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"/Users/dpadil10/ASU Dropbox/Dylan Padilla/Yale/aRanSyl-PopGen/data/climate"</span>,</span>
<span id="cb28-84"><a href="#cb28-84" aria-hidden="true" tabindex="-1"></a>                   <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb28-85"><a href="#cb28-85" aria-hidden="true" tabindex="-1"></a>file</span>
<span id="cb28-86"><a href="#cb28-86" aria-hidden="true" tabindex="-1"></a>climate_future.var <span class="ot">&lt;-</span> <span class="fu">stack</span>(file[<span class="dv">2</span>])</span>
<span id="cb28-87"><a href="#cb28-87" aria-hidden="true" tabindex="-1"></a><span class="do">##climate_future.var &lt;-climate_future.var[[c(1, 4, 12, 15)]]</span></span>
<span id="cb28-88"><a href="#cb28-88" aria-hidden="true" tabindex="-1"></a>climate_future.var</span>
<span id="cb28-89"><a href="#cb28-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-90"><a href="#cb28-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-91"><a href="#cb28-91" aria-hidden="true" tabindex="-1"></a><span class="do">## pulling out coordinates and plotting sample sites</span></span>
<span id="cb28-92"><a href="#cb28-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-93"><a href="#cb28-93" aria-hidden="true" tabindex="-1"></a>coord <span class="ot">&lt;-</span> meta[ , <span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">2</span>)]</span>
<span id="cb28-94"><a href="#cb28-94" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(coord)</span>
<span id="cb28-95"><a href="#cb28-95" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(coord)</span>
<span id="cb28-96"><a href="#cb28-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-97"><a href="#cb28-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-98"><a href="#cb28-98" aria-hidden="true" tabindex="-1"></a><span class="do">## extracting historical environmental data for A. thaliana samples</span></span>
<span id="cb28-99"><a href="#cb28-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-100"><a href="#cb28-100" aria-hidden="true" tabindex="-1"></a>X.env <span class="ot">&lt;-</span> terra<span class="sc">::</span><span class="fu">extract</span>(<span class="at">x =</span> climate.var, </span>
<span id="cb28-101"><a href="#cb28-101" aria-hidden="true" tabindex="-1"></a>                       <span class="at">y =</span> <span class="fu">data.frame</span>(coord), </span>
<span id="cb28-102"><a href="#cb28-102" aria-hidden="true" tabindex="-1"></a>                       <span class="at">cells =</span> <span class="cn">FALSE</span>)</span>
<span id="cb28-103"><a href="#cb28-103" aria-hidden="true" tabindex="-1"></a><span class="do">## remove IDs</span></span>
<span id="cb28-104"><a href="#cb28-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-105"><a href="#cb28-105" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(X.env)</span>
<span id="cb28-106"><a href="#cb28-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-107"><a href="#cb28-107" aria-hidden="true" tabindex="-1"></a><span class="do">## well use a simple approach to imputing missing genotype values:</span></span>
<span id="cb28-108"><a href="#cb28-108" aria-hidden="true" tabindex="-1"></a><span class="do">## we will impute using the most common genotype at each SNP across</span></span>
<span id="cb28-109"><a href="#cb28-109" aria-hidden="true" tabindex="-1"></a><span class="do">## all individuals</span></span>
<span id="cb28-110"><a href="#cb28-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-111"><a href="#cb28-111" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">apply</span>(aRanSyl_genind_pops<span class="sc">$</span>tab, <span class="dv">2</span>,</span>
<span id="cb28-112"><a href="#cb28-112" aria-hidden="true" tabindex="-1"></a>           <span class="cf">function</span>(x) <span class="fu">replace</span>(x, <span class="fu">is.na</span>(x),</span>
<span id="cb28-113"><a href="#cb28-113" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">as.numeric</span>(<span class="fu">names</span>(<span class="fu">which.max</span>(<span class="fu">table</span>(x))))))</span>
<span id="cb28-114"><a href="#cb28-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-115"><a href="#cb28-115" aria-hidden="true" tabindex="-1"></a><span class="do">##x &lt;- tab(aRanSyl_genind_pops, NA.method = "mean")</span></span>
<span id="cb28-116"><a href="#cb28-116" aria-hidden="true" tabindex="-1"></a>x[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb28-117"><a href="#cb28-117" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(x)</span>
<span id="cb28-118"><a href="#cb28-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-119"><a href="#cb28-119" aria-hidden="true" tabindex="-1"></a>snppca <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(x)</span>
<span id="cb28-120"><a href="#cb28-120" aria-hidden="true" tabindex="-1"></a>snppca[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span>
<span id="cb28-121"><a href="#cb28-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-122"><a href="#cb28-122" aria-hidden="true" tabindex="-1"></a><span class="do">## genotypes and environmental data are in the same order</span></span>
<span id="cb28-123"><a href="#cb28-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-124"><a href="#cb28-124" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">rownames</span>(snppca), meta[, <span class="dv">1</span>])</span>
<span id="cb28-125"><a href="#cb28-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-126"><a href="#cb28-126" aria-hidden="true" tabindex="-1"></a><span class="do">## latent factor GEA model</span></span>
<span id="cb28-127"><a href="#cb28-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-128"><a href="#cb28-128" aria-hidden="true" tabindex="-1"></a>mod_lfmm <span class="ot">&lt;-</span> LEA<span class="sc">::</span><span class="fu">lfmm2</span>(<span class="at">input =</span> snppca, </span>
<span id="cb28-129"><a href="#cb28-129" aria-hidden="true" tabindex="-1"></a>                      <span class="at">env =</span> <span class="fu">scale</span>(X.env), </span>
<span id="cb28-130"><a href="#cb28-130" aria-hidden="true" tabindex="-1"></a>                      <span class="at">K =</span> <span class="dv">5</span>, </span>
<span id="cb28-131"><a href="#cb28-131" aria-hidden="true" tabindex="-1"></a>                      <span class="at">effect.sizes =</span> <span class="cn">TRUE</span>)</span>
<span id="cb28-132"><a href="#cb28-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-133"><a href="#cb28-133" aria-hidden="true" tabindex="-1"></a><span class="do">## get environmental effect sizes</span></span>
<span id="cb28-134"><a href="#cb28-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-135"><a href="#cb28-135" aria-hidden="true" tabindex="-1"></a>B <span class="ot">&lt;-</span> mod_lfmm<span class="sc">@</span>B</span>
<span id="cb28-136"><a href="#cb28-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-137"><a href="#cb28-137" aria-hidden="true" tabindex="-1"></a>pv <span class="ot">&lt;-</span> <span class="fu">lfmm2.test</span>(mod_lfmm, </span>
<span id="cb28-138"><a href="#cb28-138" aria-hidden="true" tabindex="-1"></a>                <span class="at">input =</span> snppca, </span>
<span id="cb28-139"><a href="#cb28-139" aria-hidden="true" tabindex="-1"></a>                <span class="at">env =</span> <span class="fu">scale</span>(X.env), </span>
<span id="cb28-140"><a href="#cb28-140" aria-hidden="true" tabindex="-1"></a>                <span class="at">full =</span> <span class="cn">TRUE</span>)</span>
<span id="cb28-141"><a href="#cb28-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-142"><a href="#cb28-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-143"><a href="#cb28-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-144"><a href="#cb28-144" aria-hidden="true" tabindex="-1"></a><span class="do">## define candidate loci for GO analysis</span></span>
<span id="cb28-145"><a href="#cb28-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-146"><a href="#cb28-146" aria-hidden="true" tabindex="-1"></a>candidates <span class="ot">&lt;-</span>  <span class="sc">-</span><span class="fu">log10</span>(pv<span class="sc">$</span>pvalue) <span class="sc">&gt;</span> <span class="dv">5</span></span>
<span id="cb28-147"><a href="#cb28-147" aria-hidden="true" tabindex="-1"></a>(adaptive.snps <span class="ot">&lt;-</span> <span class="fu">colnames</span>(snppca)[<span class="fu">which</span>(<span class="sc">-</span><span class="fu">log10</span>(pv<span class="sc">$</span>pvalue) <span class="sc">&gt;</span> <span class="dv">5</span>)])</span>
<span id="cb28-148"><a href="#cb28-148" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">as.data.frame</span>(adaptive.snps),</span>
<span id="cb28-149"><a href="#cb28-149" aria-hidden="true" tabindex="-1"></a>            <span class="at">file =</span> <span class="st">"/Users/dpadil10/ASU Dropbox/Dylan Padilla/Yale/aRanSyl-PopGen/data/aRanSyl_adaptive_snps.txt"</span>,</span>
<span id="cb28-150"><a href="#cb28-150" aria-hidden="true" tabindex="-1"></a>            <span class="at">col.names =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb28-151"><a href="#cb28-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-152"><a href="#cb28-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-153"><a href="#cb28-153" aria-hidden="true" tabindex="-1"></a><span class="do">## how many candidate loci?</span></span>
<span id="cb28-154"><a href="#cb28-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-155"><a href="#cb28-155" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(candidates)</span>
<span id="cb28-156"><a href="#cb28-156" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(adaptive.snps)</span>
<span id="cb28-157"><a href="#cb28-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-158"><a href="#cb28-158" aria-hidden="true" tabindex="-1"></a><span class="do">## taking all loci for GO analysis</span></span>
<span id="cb28-159"><a href="#cb28-159" aria-hidden="true" tabindex="-1"></a><span class="do">## candidates =  -log10(pv$pvalue) &gt; 0 </span></span>
<span id="cb28-160"><a href="#cb28-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-161"><a href="#cb28-161" aria-hidden="true" tabindex="-1"></a>all.loc <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ID =</span> <span class="fu">colnames</span>(snppca),</span>
<span id="cb28-162"><a href="#cb28-162" aria-hidden="true" tabindex="-1"></a>                      <span class="at">pvalues =</span> <span class="fu">as.numeric</span>(pv<span class="sc">$</span>pvalues))</span>
<span id="cb28-163"><a href="#cb28-163" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(all.loc)</span>
<span id="cb28-164"><a href="#cb28-164" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span>(all.loc)</span>
<span id="cb28-165"><a href="#cb28-165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-166"><a href="#cb28-166" aria-hidden="true" tabindex="-1"></a>unique.all.loc <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb28-167"><a href="#cb28-167" aria-hidden="true" tabindex="-1"></a>unique.all.loc <span class="ot">&lt;-</span> all.loc[<span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">nrow</span>(all.loc), <span class="dv">2</span>), ]</span>
<span id="cb28-168"><a href="#cb28-168" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(unique.all.loc)</span>
<span id="cb28-169"><a href="#cb28-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-170"><a href="#cb28-170" aria-hidden="true" tabindex="-1"></a>unique.all.loc<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">..*"</span>, <span class="st">""</span>, unique.all.loc<span class="sc">$</span>ID)</span>
<span id="cb28-171"><a href="#cb28-171" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(unique.all.loc)</span>
<span id="cb28-172"><a href="#cb28-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-173"><a href="#cb28-173" aria-hidden="true" tabindex="-1"></a>pos.loc <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"/Users/dpadil10/ASU Dropbox/Dylan Padilla/Yale/aRanSyl-PopGen/data/aRanSyl_vcf_fix_data.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb28-174"><a href="#cb28-174" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(pos.loc)</span>
<span id="cb28-175"><a href="#cb28-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-176"><a href="#cb28-176" aria-hidden="true" tabindex="-1"></a>fix.data <span class="ot">&lt;-</span> <span class="fu">merge</span>(pos.loc, unique.all.loc, <span class="at">by =</span> <span class="st">"ID"</span>)</span>
<span id="cb28-177"><a href="#cb28-177" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(fix.data)</span>
<span id="cb28-178"><a href="#cb28-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-179"><a href="#cb28-179" aria-hidden="true" tabindex="-1"></a>gff <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"/Users/dpadil10/ASU Dropbox/Dylan Padilla/Yale/aRanSyl-PopGen/data/aRansyl_genome.gff"</span>, <span class="at">header =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb28-180"><a href="#cb28-180" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(gff)</span>
<span id="cb28-181"><a href="#cb28-181" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(gff) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"CHROM"</span>, <span class="st">"Source"</span>, <span class="st">"Type"</span>, <span class="st">"Start"</span>, <span class="st">"End"</span>,</span>
<span id="cb28-182"><a href="#cb28-182" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Score"</span>, <span class="st">"Strand"</span>, <span class="st">"Phase"</span>, <span class="st">"Attributes"</span>)</span>
<span id="cb28-183"><a href="#cb28-183" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gff)</span>
<span id="cb28-184"><a href="#cb28-184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-185"><a href="#cb28-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-186"><a href="#cb28-186" aria-hidden="true" tabindex="-1"></a>gff.genes <span class="ot">&lt;-</span> gff[gff<span class="sc">$</span>Type <span class="sc">==</span> <span class="st">"gene"</span>, ]</span>
<span id="cb28-187"><a href="#cb28-187" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(gff.genes)</span>
<span id="cb28-188"><a href="#cb28-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-189"><a href="#cb28-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-190"><a href="#cb28-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-191"><a href="#cb28-191" aria-hidden="true" tabindex="-1"></a><span class="do">## create GRanges objects</span></span>
<span id="cb28-192"><a href="#cb28-192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-193"><a href="#cb28-193" aria-hidden="true" tabindex="-1"></a>pos_gr <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(</span>
<span id="cb28-194"><a href="#cb28-194" aria-hidden="true" tabindex="-1"></a>  <span class="at">seqnames =</span> fix.data<span class="sc">$</span>CHROM,</span>
<span id="cb28-195"><a href="#cb28-195" aria-hidden="true" tabindex="-1"></a>  <span class="at">ranges =</span> <span class="fu">IRanges</span>(fix.data<span class="sc">$</span>POS, <span class="at">width =</span> <span class="dv">1</span>)</span>
<span id="cb28-196"><a href="#cb28-196" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb28-197"><a href="#cb28-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-198"><a href="#cb28-198" aria-hidden="true" tabindex="-1"></a>genes_gr <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(</span>
<span id="cb28-199"><a href="#cb28-199" aria-hidden="true" tabindex="-1"></a>  <span class="at">seqnames =</span> gff.genes<span class="sc">$</span>CHROM,</span>
<span id="cb28-200"><a href="#cb28-200" aria-hidden="true" tabindex="-1"></a>  <span class="at">ranges =</span> <span class="fu">IRanges</span>(gff.genes<span class="sc">$</span>Start, gff.genes<span class="sc">$</span>End),</span>
<span id="cb28-201"><a href="#cb28-201" aria-hidden="true" tabindex="-1"></a>  <span class="at">mcols =</span> gff.genes[ , <span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)]</span>
<span id="cb28-202"><a href="#cb28-202" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb28-203"><a href="#cb28-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-204"><a href="#cb28-204" aria-hidden="true" tabindex="-1"></a><span class="do">## find overlaps</span></span>
<span id="cb28-205"><a href="#cb28-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-206"><a href="#cb28-206" aria-hidden="true" tabindex="-1"></a>overlaps <span class="ot">&lt;-</span> <span class="fu">findOverlaps</span>(pos_gr, genes_gr)</span>
<span id="cb28-207"><a href="#cb28-207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-208"><a href="#cb28-208" aria-hidden="true" tabindex="-1"></a><span class="do">## create the resulting table efficiently</span></span>
<span id="cb28-209"><a href="#cb28-209" aria-hidden="true" tabindex="-1"></a><span class="do">## this joins the original position data with</span></span>
<span id="cb28-210"><a href="#cb28-210" aria-hidden="true" tabindex="-1"></a><span class="do">## the gene data for matching entries</span></span>
<span id="cb28-211"><a href="#cb28-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-212"><a href="#cb28-212" aria-hidden="true" tabindex="-1"></a>gene.table.optimized <span class="ot">&lt;-</span> <span class="fu">cbind</span>(</span>
<span id="cb28-213"><a href="#cb28-213" aria-hidden="true" tabindex="-1"></a>  fix.data[<span class="fu">queryHits</span>(overlaps), ],</span>
<span id="cb28-214"><a href="#cb28-214" aria-hidden="true" tabindex="-1"></a>  gff.genes[<span class="fu">subjectHits</span>(overlaps), <span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="dv">6</span><span class="sc">:</span><span class="dv">8</span>)]</span>
<span id="cb28-215"><a href="#cb28-215" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb28-216"><a href="#cb28-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-217"><a href="#cb28-217" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(gene.table.optimized)</span>
<span id="cb28-218"><a href="#cb28-218" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gene.table.optimized)</span>
<span id="cb28-219"><a href="#cb28-219" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-220"><a href="#cb28-220" aria-hidden="true" tabindex="-1"></a>id.genes <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">";.*"</span>, <span class="st">""</span>, gene.table.optimized<span class="sc">$</span>Attributes)</span>
<span id="cb28-221"><a href="#cb28-221" aria-hidden="true" tabindex="-1"></a>id.genes <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"ID="</span>, <span class="st">""</span>, id.genes)</span>
<span id="cb28-222"><a href="#cb28-222" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(id.genes)</span>
<span id="cb28-223"><a href="#cb28-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-224"><a href="#cb28-224" aria-hidden="true" tabindex="-1"></a>name.genes <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"^.*;"</span>, <span class="st">""</span>, gene.table.optimized<span class="sc">$</span>Attributes)</span>
<span id="cb28-225"><a href="#cb28-225" aria-hidden="true" tabindex="-1"></a>name.genes <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Name="</span>, <span class="st">""</span>, name.genes)</span>
<span id="cb28-226"><a href="#cb28-226" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(name.genes)</span>
<span id="cb28-227"><a href="#cb28-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-228"><a href="#cb28-228" aria-hidden="true" tabindex="-1"></a>gene.table.optimized<span class="sc">$</span>geneID <span class="ot">&lt;-</span> id.genes</span>
<span id="cb28-229"><a href="#cb28-229" aria-hidden="true" tabindex="-1"></a>gene.table.optimized<span class="sc">$</span>geneName <span class="ot">&lt;-</span> name.genes</span>
<span id="cb28-230"><a href="#cb28-230" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gene.table.optimized)</span>
<span id="cb28-231"><a href="#cb28-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-232"><a href="#cb28-232" aria-hidden="true" tabindex="-1"></a><span class="do">## GO enrichment</span></span>
<span id="cb28-233"><a href="#cb28-233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-234"><a href="#cb28-234" aria-hidden="true" tabindex="-1"></a>gene.table.GO <span class="ot">&lt;-</span> gene.table.optimized[<span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">9</span>, <span class="dv">14</span>, <span class="dv">15</span>)]</span>
<span id="cb28-235"><a href="#cb28-235" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gene.table.GO)</span>
<span id="cb28-236"><a href="#cb28-236" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-237"><a href="#cb28-237" aria-hidden="true" tabindex="-1"></a>gene.table.GO <span class="ot">&lt;-</span> gene.table.GO[<span class="sc">!</span><span class="fu">duplicated</span>(gene.table.GO<span class="sc">$</span>geneName), ]</span>
<span id="cb28-238"><a href="#cb28-238" aria-hidden="true" tabindex="-1"></a>gene.table.GO<span class="sc">$</span>geneName <span class="ot">&lt;-</span>  <span class="fu">gsub</span>(<span class="st">"ID="</span>, <span class="st">""</span>, gene.table.GO<span class="sc">$</span>geneName)</span>
<span id="cb28-239"><a href="#cb28-239" aria-hidden="true" tabindex="-1"></a>gene.table.GO<span class="sc">$</span>geneName <span class="ot">&lt;-</span>  <span class="fu">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>, gene.table.GO<span class="sc">$</span>geneName)</span>
<span id="cb28-240"><a href="#cb28-240" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gene.table.GO)</span>
<span id="cb28-241"><a href="#cb28-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-242"><a href="#cb28-242" aria-hidden="true" tabindex="-1"></a>gene.table.GO<span class="sc">$</span>geneID <span class="ot">&lt;-</span>  <span class="fu">gsub</span>(<span class="st">"aRanSyl1.0_"</span>, <span class="st">""</span>, gene.table.GO<span class="sc">$</span>geneID)</span>
<span id="cb28-243"><a href="#cb28-243" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gene.table.GO)</span>
<span id="cb28-244"><a href="#cb28-244" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(gene.table.GO)</span>
<span id="cb28-245"><a href="#cb28-245" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(gene.table.GO, <span class="at">file =</span> <span class="st">"/Users/dpadil10/ASU Dropbox/Dylan Padilla/Yale/aRanSyl-PopGen/data/aRanSyl_gene_table.csv"</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb28-246"><a href="#cb28-246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-247"><a href="#cb28-247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-248"><a href="#cb28-248" aria-hidden="true" tabindex="-1"></a>EntrezID_genes_cleaned <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(EntrezID_genes[<span class="sc">!</span><span class="fu">is.na</span>(EntrezID_genes)])</span>
<span id="cb28-249"><a href="#cb28-249" aria-hidden="true" tabindex="-1"></a>EntrezID_genes_cleaned <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(EntrezID_genes_cleaned)</span>
<span id="cb28-250"><a href="#cb28-250" aria-hidden="true" tabindex="-1"></a>EntrezID_genes_cleaned<span class="sc">$</span>geneName <span class="ot">&lt;-</span> <span class="fu">row.names</span>(EntrezID_genes_cleaned)</span>
<span id="cb28-251"><a href="#cb28-251" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(EntrezID_genes_cleaned)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">"geneID"</span></span>
<span id="cb28-252"><a href="#cb28-252" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(EntrezID_genes_cleaned)</span>
<span id="cb28-253"><a href="#cb28-253" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(EntrezID_genes_cleaned)</span>
<span id="cb28-254"><a href="#cb28-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-255"><a href="#cb28-255" aria-hidden="true" tabindex="-1"></a>EntrezID_gene_table <span class="ot">&lt;-</span> <span class="fu">merge</span>(gene.table.GO[<span class="sc">-</span><span class="dv">4</span>],</span>
<span id="cb28-256"><a href="#cb28-256" aria-hidden="true" tabindex="-1"></a>                             EntrezID_genes_cleaned,</span>
<span id="cb28-257"><a href="#cb28-257" aria-hidden="true" tabindex="-1"></a>                             <span class="at">by =</span> <span class="st">"geneName"</span>)</span>
<span id="cb28-258"><a href="#cb28-258" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(EntrezID_gene_table)</span>
<span id="cb28-259"><a href="#cb28-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-260"><a href="#cb28-260" aria-hidden="true" tabindex="-1"></a>geneList <span class="ot">&lt;-</span> <span class="fu">setNames</span>(EntrezID_gene_table<span class="sc">$</span>pvalues,</span>
<span id="cb28-261"><a href="#cb28-261" aria-hidden="true" tabindex="-1"></a>                     EntrezID_gene_table<span class="sc">$</span>geneID)</span>
<span id="cb28-262"><a href="#cb28-262" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(geneList)</span>
<span id="cb28-263"><a href="#cb28-263" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(geneList)</span>
<span id="cb28-264"><a href="#cb28-264" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-265"><a href="#cb28-265" aria-hidden="true" tabindex="-1"></a><span class="do">## I am using the Genome wide annotation for the mouse</span></span>
<span id="cb28-266"><a href="#cb28-266" aria-hidden="true" tabindex="-1"></a><span class="do">## published on Bioconductor</span></span>
<span id="cb28-267"><a href="#cb28-267" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-268"><a href="#cb28-268" aria-hidden="true" tabindex="-1"></a>GOdata <span class="ot">&lt;-</span> <span class="fu">new</span>(<span class="st">"topGOdata"</span>,</span>
<span id="cb28-269"><a href="#cb28-269" aria-hidden="true" tabindex="-1"></a>    <span class="at">ontology =</span> <span class="st">"BP"</span>,</span>
<span id="cb28-270"><a href="#cb28-270" aria-hidden="true" tabindex="-1"></a>    <span class="at">allGenes =</span> geneList,</span>
<span id="cb28-271"><a href="#cb28-271" aria-hidden="true" tabindex="-1"></a>    <span class="at">geneSelectionFun =</span> <span class="cf">function</span>(x) (<span class="sc">-</span><span class="fu">log10</span>(x) <span class="sc">&gt;</span> <span class="dv">5</span>),</span>
<span id="cb28-272"><a href="#cb28-272" aria-hidden="true" tabindex="-1"></a>    <span class="at">annot =</span> annFUN.org , <span class="at">mapping =</span> <span class="st">"org.Mm.eg.db"</span>)</span>
<span id="cb28-273"><a href="#cb28-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-274"><a href="#cb28-274" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-275"><a href="#cb28-275" aria-hidden="true" tabindex="-1"></a><span class="fu">graph</span>(GOdata)</span>
<span id="cb28-276"><a href="#cb28-276" aria-hidden="true" tabindex="-1"></a>test.stat <span class="ot">&lt;-</span> <span class="fu">new</span>(<span class="st">"classicCount"</span>, <span class="at">testStatistic =</span> GOFisherTest,</span>
<span id="cb28-277"><a href="#cb28-277" aria-hidden="true" tabindex="-1"></a>                 <span class="at">name =</span> <span class="st">"Fisher test"</span>)</span>
<span id="cb28-278"><a href="#cb28-278" aria-hidden="true" tabindex="-1"></a>resultFisher <span class="ot">&lt;-</span> <span class="fu">getSigGroups</span>(GOdata, test.stat)</span>
<span id="cb28-279"><a href="#cb28-279" aria-hidden="true" tabindex="-1"></a>resultFisher</span>
<span id="cb28-280"><a href="#cb28-280" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-281"><a href="#cb28-281" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">cex.lab =</span> <span class="fl">0.7</span>) <span class="co"># Adjust font size</span></span>
<span id="cb28-282"><a href="#cb28-282" aria-hidden="true" tabindex="-1"></a><span class="fu">showSigOfNodes</span>(GOdata, </span>
<span id="cb28-283"><a href="#cb28-283" aria-hidden="true" tabindex="-1"></a>               <span class="fu">score</span>(resultFisher), </span>
<span id="cb28-284"><a href="#cb28-284" aria-hidden="true" tabindex="-1"></a>               <span class="at">firstSigNodes =</span> <span class="dv">5</span>, <span class="co"># Number of top significant nodes to display</span></span>
<span id="cb28-285"><a href="#cb28-285" aria-hidden="true" tabindex="-1"></a>               <span class="at">useInfo =</span> <span class="st">'def'</span>,  <span class="co"># Display definition</span></span>
<span id="cb28-286"><a href="#cb28-286" aria-hidden="true" tabindex="-1"></a>               <span class="at">swPlot =</span> <span class="cn">TRUE</span>)</span>
<span id="cb28-287"><a href="#cb28-287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-288"><a href="#cb28-288" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-289"><a href="#cb28-289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-290"><a href="#cb28-290" aria-hidden="true" tabindex="-1"></a><span class="do">## perform GO enrichment analysis</span></span>
<span id="cb28-291"><a href="#cb28-291" aria-hidden="true" tabindex="-1"></a><span class="do">## I am using the Genome wide annotation for the Mouse published on Bioconductor</span></span>
<span id="cb28-292"><a href="#cb28-292" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-293"><a href="#cb28-293" aria-hidden="true" tabindex="-1"></a>ego <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene =</span> EntrezID_gene_table<span class="sc">$</span>geneName,</span>
<span id="cb28-294"><a href="#cb28-294" aria-hidden="true" tabindex="-1"></a>                <span class="at">OrgDb =</span> org.Mm.eg.db,</span>
<span id="cb28-295"><a href="#cb28-295" aria-hidden="true" tabindex="-1"></a>                <span class="at">keyType =</span> <span class="st">"SYMBOL"</span>, <span class="co"># Or "ENTREZID", "ENSEMBL"</span></span>
<span id="cb28-296"><a href="#cb28-296" aria-hidden="true" tabindex="-1"></a>                <span class="at">ont =</span> <span class="st">"BP"</span>, <span class="co"># Biological Process, CC (Cellular Component), MF (Molecular Function)</span></span>
<span id="cb28-297"><a href="#cb28-297" aria-hidden="true" tabindex="-1"></a>                <span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>,</span>
<span id="cb28-298"><a href="#cb28-298" aria-hidden="true" tabindex="-1"></a>                <span class="at">qvalueCutoff =</span> <span class="fl">0.05</span>)</span>
<span id="cb28-299"><a href="#cb28-299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-300"><a href="#cb28-300" aria-hidden="true" tabindex="-1"></a><span class="do">## view results</span></span>
<span id="cb28-301"><a href="#cb28-301" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-302"><a href="#cb28-302" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ego)</span>
<span id="cb28-303"><a href="#cb28-303" aria-hidden="true" tabindex="-1"></a>ego<span class="sc">$</span>Description</span>
<span id="cb28-304"><a href="#cb28-304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-305"><a href="#cb28-305" aria-hidden="true" tabindex="-1"></a>selected.go <span class="ot">&lt;-</span> ego[<span class="fu">c</span>(<span class="dv">119</span>, <span class="dv">123</span>, <span class="dv">348</span>, <span class="dv">467</span>, <span class="dv">557</span>, <span class="dv">385</span>), ]</span>
<span id="cb28-306"><a href="#cb28-306" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-307"><a href="#cb28-307" aria-hidden="true" tabindex="-1"></a>ego.table <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(ego)</span>
<span id="cb28-308"><a href="#cb28-308" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ego.table)</span>
<span id="cb28-309"><a href="#cb28-309" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-310"><a href="#cb28-310" aria-hidden="true" tabindex="-1"></a>terms <span class="ot">&lt;-</span> ego.table[<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">11</span>, <span class="dv">9</span>)]</span>
<span id="cb28-311"><a href="#cb28-311" aria-hidden="true" tabindex="-1"></a>terms<span class="sc">$</span>Category <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="st">"BP"</span>, <span class="fu">nrow</span>(terms))</span>
<span id="cb28-312"><a href="#cb28-312" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(terms)[<span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">4</span>)] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Term"</span>, <span class="st">"Genes"</span>, <span class="st">"adj_pval"</span>)</span>
<span id="cb28-313"><a href="#cb28-313" aria-hidden="true" tabindex="-1"></a>terms <span class="ot">&lt;-</span> terms[<span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">4</span>)]</span>
<span id="cb28-314"><a href="#cb28-314" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-315"><a href="#cb28-315" aria-hidden="true" tabindex="-1"></a><span class="do">## setting up data for plotting manhattan plot</span></span>
<span id="cb28-316"><a href="#cb28-316" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-317"><a href="#cb28-317" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(fix.data)</span>
<span id="cb28-318"><a href="#cb28-318" aria-hidden="true" tabindex="-1"></a>man <span class="ot">&lt;-</span> fix.data[<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">9</span>)]</span>
<span id="cb28-319"><a href="#cb28-319" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(man) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"SNP"</span>, <span class="st">"CHR"</span>, <span class="st">"BP"</span>, <span class="st">"P"</span>)</span>
<span id="cb28-320"><a href="#cb28-320" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(man)</span>
<span id="cb28-321"><a href="#cb28-321" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(man)</span>
<span id="cb28-322"><a href="#cb28-322" aria-hidden="true" tabindex="-1"></a>man<span class="sc">$</span>CHR <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">factor</span>(man<span class="sc">$</span>CHR))</span>
<span id="cb28-323"><a href="#cb28-323" aria-hidden="true" tabindex="-1"></a>man <span class="ot">&lt;-</span> man[man<span class="sc">$</span>CHR <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">13</span>), ]</span>
<span id="cb28-324"><a href="#cb28-324" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-325"><a href="#cb28-325" aria-hidden="true" tabindex="-1"></a><span class="do">## plotting manhattan plot</span></span>
<span id="cb28-326"><a href="#cb28-326" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-327"><a href="#cb28-327" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mgp =</span> <span class="fu">c</span>(<span class="fl">2.7</span>, <span class="dv">1</span>, <span class="dv">0</span>))</span>
<span id="cb28-328"><a href="#cb28-328" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(man, <span class="at">chr=</span><span class="st">"CHR"</span>, <span class="at">bp=</span><span class="st">"BP"</span>, <span class="at">snp=</span><span class="st">"SNP"</span>, <span class="at">p=</span><span class="st">"P"</span>, <span class="at">xaxt =</span> <span class="st">"n"</span>,</span>
<span id="cb28-329"><a href="#cb28-329" aria-hidden="true" tabindex="-1"></a>          <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">60</span>), <span class="at">type =</span> <span class="st">"n"</span>, <span class="at">cex.axis =</span> <span class="fl">0.8</span>, <span class="at">cex =</span> <span class="fl">0.5</span>)</span>
<span id="cb28-330"><a href="#cb28-330" aria-hidden="true" tabindex="-1"></a><span class="fu">grid</span>()</span>
<span id="cb28-331"><a href="#cb28-331" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">new =</span> <span class="cn">TRUE</span>)</span>
<span id="cb28-332"><a href="#cb28-332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-333"><a href="#cb28-333" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(man, <span class="at">chr=</span><span class="st">"CHR"</span>, <span class="at">bp=</span><span class="st">"BP"</span>, <span class="at">snp=</span><span class="st">"SNP"</span>, <span class="at">p=</span><span class="st">"P"</span>, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">60</span>),</span>
<span id="cb28-334"><a href="#cb28-334" aria-hidden="true" tabindex="-1"></a>          <span class="at">yaxt =</span> <span class="st">"n"</span>, <span class="at">ylab =</span> <span class="st">""</span>, <span class="at">xlab =</span> <span class="st">""</span>, <span class="at">cex.axis =</span> <span class="fl">0.6</span>, <span class="at">cex =</span> <span class="fl">0.5</span>)</span>
<span id="cb28-335"><a href="#cb28-335" aria-hidden="true" tabindex="-1"></a><span class="fu">box</span>()</span>
<span id="cb28-336"><a href="#cb28-336" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-337"><a href="#cb28-337" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">recordPlot</span>()</span>
<span id="cb28-338"><a href="#cb28-338" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-339"><a href="#cb28-339" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">barplot</span>(ego, <span class="at">showCategory =</span> selected.go<span class="sc">$</span>Description, <span class="at">label_format =</span> <span class="dv">80</span>)</span>
<span id="cb28-340"><a href="#cb28-340" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-341"><a href="#cb28-341" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(p1, p2, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">labels =</span> <span class="st">"AUTO"</span>, <span class="at">label_fontface =</span> <span class="st">"plain"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><img src="imgs/aRanSyl_adaptive_snps.png" class="img-fluid"></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="do">## nc = resolution, higher is better but slower </span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>nc <span class="ot">=</span> <span class="dv">200</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a><span class="do">## range of longitude for North America</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>long.mat <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">180</span>, <span class="sc">-</span><span class="dv">50</span>, <span class="at">length =</span> nc)</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a><span class="do">## range of latitude for North America</span></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>lat.mat <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">25</span>, <span class="dv">70</span>, <span class="at">length =</span> nc)</span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a><span class="do">## matrix of cells for Europe (nc times nc)</span></span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a>coord.mat <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (x <span class="cf">in</span> long.mat) </span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (y <span class="cf">in</span> lat.mat) coord.mat <span class="ot">&lt;-</span> <span class="fu">rbind</span>(coord.mat, <span class="fu">c</span>(x,y))</span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a><span class="do">## extract historical climate</span></span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a>env.new <span class="ot">&lt;-</span> terra<span class="sc">::</span><span class="fu">extract</span>(<span class="at">x =</span> climate.var, </span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a>                         <span class="at">y =</span> <span class="fu">data.frame</span>(coord.mat), </span>
<span id="cb29-24"><a href="#cb29-24" aria-hidden="true" tabindex="-1"></a>                         <span class="at">cells =</span> <span class="cn">FALSE</span>)</span>
<span id="cb29-25"><a href="#cb29-25" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(env.new)</span>
<span id="cb29-26"><a href="#cb29-26" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span>(env.new)</span>
<span id="cb29-27"><a href="#cb29-27" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(env.new)</span>
<span id="cb29-28"><a href="#cb29-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-29"><a href="#cb29-29" aria-hidden="true" tabindex="-1"></a><span class="do">## extract future climate</span></span>
<span id="cb29-30"><a href="#cb29-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-31"><a href="#cb29-31" aria-hidden="true" tabindex="-1"></a>env.pred <span class="ot">&lt;-</span> terra<span class="sc">::</span><span class="fu">extract</span>(<span class="at">x =</span> climate_future.var, </span>
<span id="cb29-32"><a href="#cb29-32" aria-hidden="true" tabindex="-1"></a>                          <span class="at">y =</span> <span class="fu">data.frame</span>(coord.mat), </span>
<span id="cb29-33"><a href="#cb29-33" aria-hidden="true" tabindex="-1"></a>                          <span class="at">cells =</span> <span class="cn">FALSE</span>)</span>
<span id="cb29-34"><a href="#cb29-34" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(env.pred)</span>
<span id="cb29-35"><a href="#cb29-35" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span>(env.pred)</span>
<span id="cb29-36"><a href="#cb29-36" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(env.pred)</span>
<span id="cb29-37"><a href="#cb29-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-38"><a href="#cb29-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-39"><a href="#cb29-39" aria-hidden="true" tabindex="-1"></a><span class="do">## scaling bioclimatic variables (with the same scale as in the lfmm)</span></span>
<span id="cb29-40"><a href="#cb29-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-41"><a href="#cb29-41" aria-hidden="true" tabindex="-1"></a>m.x <span class="ot">&lt;-</span> <span class="fu">apply</span>(X.env, <span class="dv">2</span>, <span class="at">FUN =</span> <span class="cf">function</span>(x) <span class="fu">mean</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb29-42"><a href="#cb29-42" aria-hidden="true" tabindex="-1"></a>sd.x <span class="ot">&lt;-</span> <span class="fu">apply</span>(X.env, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">sd</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb29-43"><a href="#cb29-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-44"><a href="#cb29-44" aria-hidden="true" tabindex="-1"></a>env.new <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">t</span>(env.new) <span class="sc">-</span> m.x) <span class="sc">%*%</span> <span class="fu">diag</span>(<span class="dv">1</span><span class="sc">/</span>sd.x)</span>
<span id="cb29-45"><a href="#cb29-45" aria-hidden="true" tabindex="-1"></a>env.pred <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">t</span>(env.pred) <span class="sc">-</span> m.x) <span class="sc">%*%</span> <span class="fu">diag</span>(<span class="dv">1</span><span class="sc">/</span>sd.x)</span>
<span id="cb29-46"><a href="#cb29-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-47"><a href="#cb29-47" aria-hidden="true" tabindex="-1"></a><span class="do">## geometric genomic offset for a specific long and lat</span></span>
<span id="cb29-48"><a href="#cb29-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-49"><a href="#cb29-49" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(((env.new <span class="sc">-</span> env.pred)[<span class="dv">36139</span>,]  <span class="sc">%*%</span> <span class="fu">t</span>(B[candidates,]))<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb29-50"><a href="#cb29-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-51"><a href="#cb29-51" aria-hidden="true" tabindex="-1"></a><span class="do">## gg contains the Gain et al. geometric GO computed at each matrix cell</span></span>
<span id="cb29-52"><a href="#cb29-52" aria-hidden="true" tabindex="-1"></a><span class="do">## be patient, it may be very slow for large nc</span></span>
<span id="cb29-53"><a href="#cb29-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-54"><a href="#cb29-54" aria-hidden="true" tabindex="-1"></a>gg <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb29-55"><a href="#cb29-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-56"><a href="#cb29-56" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(env.new)){</span>
<span id="cb29-57"><a href="#cb29-57" aria-hidden="true" tabindex="-1"></a>    gg[i] <span class="ot">&lt;-</span> <span class="fu">mean</span>(((env.new <span class="sc">-</span> env.pred)[i,]  <span class="sc">%*%</span> <span class="fu">t</span>(B[candidates,]))<span class="sc">^</span><span class="dv">2</span>,</span>
<span id="cb29-58"><a href="#cb29-58" aria-hidden="true" tabindex="-1"></a>                  <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb29-59"><a href="#cb29-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-60"><a href="#cb29-60" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb29-61"><a href="#cb29-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-62"><a href="#cb29-62" aria-hidden="true" tabindex="-1"></a><span class="do">## matrix of genomic offset for the Europe map</span></span>
<span id="cb29-63"><a href="#cb29-63" aria-hidden="true" tabindex="-1"></a><span class="do">## NA when below sea level.</span></span>
<span id="cb29-64"><a href="#cb29-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-65"><a href="#cb29-65" aria-hidden="true" tabindex="-1"></a>go <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">matrix</span>(gg, <span class="at">byrow =</span>  <span class="cn">FALSE</span>, <span class="at">ncol =</span> nc))</span>
<span id="cb29-66"><a href="#cb29-66" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(go)</span>
<span id="cb29-67"><a href="#cb29-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-68"><a href="#cb29-68" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(<span class="fu">as.numeric</span>(go), </span>
<span id="cb29-69"><a href="#cb29-69" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">"Histogram of GO values"</span>,</span>
<span id="cb29-70"><a href="#cb29-70" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">"Geometric GO"</span>,</span>
<span id="cb29-71"><a href="#cb29-71" aria-hidden="true" tabindex="-1"></a>     <span class="at">las =</span> <span class="dv">1</span>)</span>
<span id="cb29-72"><a href="#cb29-72" aria-hidden="true" tabindex="-1"></a><span class="fu">box</span>()</span>
<span id="cb29-73"><a href="#cb29-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-74"><a href="#cb29-74" aria-hidden="true" tabindex="-1"></a><span class="fu">layout</span>(<span class="fu">matrix</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">0</span>,</span>
<span id="cb29-75"><a href="#cb29-75" aria-hidden="true" tabindex="-1"></a>                <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">0</span>,</span>
<span id="cb29-76"><a href="#cb29-76" aria-hidden="true" tabindex="-1"></a>                <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>,</span>
<span id="cb29-77"><a href="#cb29-77" aria-hidden="true" tabindex="-1"></a>                <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>), <span class="at">ncol =</span> <span class="dv">6</span>, <span class="at">nrow =</span> <span class="dv">4</span>, <span class="at">byrow =</span> <span class="cn">TRUE</span>))</span>
<span id="cb29-78"><a href="#cb29-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-79"><a href="#cb29-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-80"><a href="#cb29-80" aria-hidden="true" tabindex="-1"></a>sample.size <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">rep</span>(<span class="st">"(n = "</span>, <span class="dv">9</span>), <span class="fu">summary</span>(meta<span class="sc">$</span>Pops), <span class="st">")"</span>,</span>
<span id="cb29-81"><a href="#cb29-81" aria-hidden="true" tabindex="-1"></a>                     <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb29-82"><a href="#cb29-82" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(barplot.data) <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">rownames</span>(barplot.data),</span>
<span id="cb29-83"><a href="#cb29-83" aria-hidden="true" tabindex="-1"></a>                                sample.size, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb29-84"><a href="#cb29-84" aria-hidden="true" tabindex="-1"></a>barplot.data</span>
<span id="cb29-85"><a href="#cb29-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-86"><a href="#cb29-86" aria-hidden="true" tabindex="-1"></a><span class="do">## plotting Ne values per population</span></span>
<span id="cb29-87"><a href="#cb29-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-88"><a href="#cb29-88" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="fl">5.1</span>, <span class="fl">6.1</span>, <span class="fl">4.1</span>, <span class="fl">2.1</span>))</span>
<span id="cb29-89"><a href="#cb29-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-90"><a href="#cb29-90" aria-hidden="true" tabindex="-1"></a>bp <span class="ot">&lt;-</span> <span class="fu">barplot</span>(<span class="fu">t</span>(barplot.data), <span class="at">beside =</span> <span class="cn">TRUE</span>, <span class="at">col =</span> <span class="st">"white"</span>,</span>
<span id="cb29-91"><a href="#cb29-91" aria-hidden="true" tabindex="-1"></a>              <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">130</span>), <span class="at">xlab =</span> <span class="st">""</span>, <span class="at">ylab =</span> <span class="st">""</span>, <span class="at">las =</span> <span class="dv">1</span>,</span>
<span id="cb29-92"><a href="#cb29-92" aria-hidden="true" tabindex="-1"></a>              <span class="at">xaxt =</span> <span class="st">"n"</span>, <span class="at">yaxt =</span> <span class="st">"n"</span>, <span class="at">border =</span> <span class="cn">FALSE</span>)</span>
<span id="cb29-93"><a href="#cb29-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-94"><a href="#cb29-94" aria-hidden="true" tabindex="-1"></a><span class="fu">grid</span>()</span>
<span id="cb29-95"><a href="#cb29-95" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">new =</span> <span class="cn">TRUE</span>)</span>
<span id="cb29-96"><a href="#cb29-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-97"><a href="#cb29-97" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="fl">5.1</span>, <span class="fl">6.1</span>, <span class="fl">4.1</span>, <span class="fl">2.1</span>))</span>
<span id="cb29-98"><a href="#cb29-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-99"><a href="#cb29-99" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(<span class="fu">t</span>(barplot.data), <span class="at">beside =</span> <span class="cn">TRUE</span>,</span>
<span id="cb29-100"><a href="#cb29-100" aria-hidden="true" tabindex="-1"></a>        <span class="at">col =</span> <span class="fu">rep</span>(RColorBrewer<span class="sc">::</span><span class="fu">brewer.pal</span>(<span class="dv">9</span>,<span class="st">"Paired"</span>)[<span class="fu">factor</span>(<span class="fu">colnames</span>(<span class="fu">t</span>(barplot.data)))],</span>
<span id="cb29-101"><a href="#cb29-101" aria-hidden="true" tabindex="-1"></a>                  <span class="at">each =</span> <span class="dv">3</span>),</span>
<span id="cb29-102"><a href="#cb29-102" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">130</span>),</span>
<span id="cb29-103"><a href="#cb29-103" aria-hidden="true" tabindex="-1"></a>        <span class="at">density =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">40</span>), <span class="dv">9</span>),</span>
<span id="cb29-104"><a href="#cb29-104" aria-hidden="true" tabindex="-1"></a>        <span class="at">border =</span> <span class="cn">TRUE</span>,</span>
<span id="cb29-105"><a href="#cb29-105" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylab =</span> <span class="fu">expression</span>(<span class="st">"Contemporary"</span><span class="sc">~</span><span class="fu">italic</span>((N[e]))),</span>
<span id="cb29-106"><a href="#cb29-106" aria-hidden="true" tabindex="-1"></a>        <span class="at">las =</span> <span class="dv">1</span>, <span class="at">cex.lab =</span> <span class="fl">1.5</span>,</span>
<span id="cb29-107"><a href="#cb29-107" aria-hidden="true" tabindex="-1"></a>        <span class="at">xaxt =</span> <span class="st">"n"</span>)</span>
<span id="cb29-108"><a href="#cb29-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-109"><a href="#cb29-109" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">new =</span> <span class="cn">TRUE</span>)</span>
<span id="cb29-110"><a href="#cb29-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-111"><a href="#cb29-111" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(<span class="fu">t</span>(barplot.data), <span class="at">beside =</span> <span class="cn">TRUE</span>,</span>
<span id="cb29-112"><a href="#cb29-112" aria-hidden="true" tabindex="-1"></a>        <span class="at">col =</span> <span class="fu">alpha</span>(<span class="fu">rep</span>(RColorBrewer<span class="sc">::</span><span class="fu">brewer.pal</span>(<span class="dv">9</span>,<span class="st">"Paired"</span>)[<span class="fu">factor</span>(<span class="fu">colnames</span>(<span class="fu">t</span>(barplot.data)))],</span>
<span id="cb29-113"><a href="#cb29-113" aria-hidden="true" tabindex="-1"></a>                        <span class="at">each =</span> <span class="dv">3</span>), <span class="fl">0.5</span>),</span>
<span id="cb29-114"><a href="#cb29-114" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">130</span>),</span>
<span id="cb29-115"><a href="#cb29-115" aria-hidden="true" tabindex="-1"></a>        <span class="at">border =</span> <span class="cn">FALSE</span>,</span>
<span id="cb29-116"><a href="#cb29-116" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylab =</span> <span class="st">""</span>,</span>
<span id="cb29-117"><a href="#cb29-117" aria-hidden="true" tabindex="-1"></a>        <span class="at">xlab =</span> <span class="st">""</span>,</span>
<span id="cb29-118"><a href="#cb29-118" aria-hidden="true" tabindex="-1"></a>        <span class="at">xaxt =</span> <span class="st">"n"</span>,</span>
<span id="cb29-119"><a href="#cb29-119" aria-hidden="true" tabindex="-1"></a>        <span class="at">yaxt =</span> <span class="st">"n"</span>,</span>
<span id="cb29-120"><a href="#cb29-120" aria-hidden="true" tabindex="-1"></a>        <span class="at">las =</span> <span class="dv">1</span>)</span>
<span id="cb29-121"><a href="#cb29-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-122"><a href="#cb29-122" aria-hidden="true" tabindex="-1"></a><span class="fu">box</span>()</span>
<span id="cb29-123"><a href="#cb29-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-124"><a href="#cb29-124" aria-hidden="true" tabindex="-1"></a>idx <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb29-125"><a href="#cb29-125" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq</span>(<span class="dv">2</span>, <span class="dv">27</span>, <span class="dv">3</span>)){</span>
<span id="cb29-126"><a href="#cb29-126" aria-hidden="true" tabindex="-1"></a>    <span class="fu">axis</span>(<span class="dv">1</span>, <span class="at">at =</span> bp[i], <span class="at">labels =</span> <span class="fu">rownames</span>(barplot.data)[idx],</span>
<span id="cb29-127"><a href="#cb29-127" aria-hidden="true" tabindex="-1"></a>         <span class="at">padj =</span> <span class="fl">0.5</span>, <span class="at">cex.axis =</span> <span class="fl">0.8</span>)</span>
<span id="cb29-128"><a href="#cb29-128" aria-hidden="true" tabindex="-1"></a>    idx <span class="ot">=</span> idx <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb29-129"><a href="#cb29-129" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb29-130"><a href="#cb29-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-131"><a href="#cb29-131" aria-hidden="true" tabindex="-1"></a>ci <span class="ot">&lt;-</span> <span class="fu">aggregate</span>(<span class="fu">cbind</span>(CI_lower_0<span class="fl">.05</span>, CI_upper_0<span class="fl">.05</span>, CI_lower_0<span class="fl">.02</span>,</span>
<span id="cb29-132"><a href="#cb29-132" aria-hidden="true" tabindex="-1"></a>                      CI_upper_0<span class="fl">.02</span>, CI_lower_0<span class="fl">.01</span>, CI_upper_0<span class="fl">.01</span>)<span class="sc">~</span>Pops,</span>
<span id="cb29-133"><a href="#cb29-133" aria-hidden="true" tabindex="-1"></a>                <span class="at">data=</span>ci.data , print)</span>
<span id="cb29-134"><a href="#cb29-134" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(ci) <span class="ot">&lt;-</span> ci[,<span class="dv">1</span>]</span>
<span id="cb29-135"><a href="#cb29-135" aria-hidden="true" tabindex="-1"></a>ci <span class="ot">&lt;-</span> ci[, <span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb29-136"><a href="#cb29-136" aria-hidden="true" tabindex="-1"></a>ci</span>
<span id="cb29-137"><a href="#cb29-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-138"><a href="#cb29-138" aria-hidden="true" tabindex="-1"></a>CIs <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">abs</span>(barplot.data[ , <span class="dv">1</span>]<span class="sc">-</span>ci[ , <span class="dv">1</span>]),</span>
<span id="cb29-139"><a href="#cb29-139" aria-hidden="true" tabindex="-1"></a>             <span class="fu">abs</span>(barplot.data[ , <span class="dv">1</span>]<span class="sc">-</span>ci[ , <span class="dv">2</span>]),</span>
<span id="cb29-140"><a href="#cb29-140" aria-hidden="true" tabindex="-1"></a>             <span class="fu">abs</span>(barplot.data[ , <span class="dv">2</span>]<span class="sc">-</span>ci[ , <span class="dv">3</span>]),</span>
<span id="cb29-141"><a href="#cb29-141" aria-hidden="true" tabindex="-1"></a>             <span class="fu">abs</span>(barplot.data[ , <span class="dv">2</span>]<span class="sc">-</span>ci[ , <span class="dv">4</span>]),</span>
<span id="cb29-142"><a href="#cb29-142" aria-hidden="true" tabindex="-1"></a>             <span class="fu">abs</span>(barplot.data[ , <span class="dv">3</span>]<span class="sc">-</span>ci[ , <span class="dv">5</span>]),</span>
<span id="cb29-143"><a href="#cb29-143" aria-hidden="true" tabindex="-1"></a>             <span class="fu">abs</span>(barplot.data[ , <span class="dv">3</span>]<span class="sc">-</span>ci[ , <span class="dv">6</span>]))</span>
<span id="cb29-144"><a href="#cb29-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-145"><a href="#cb29-145" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(CIs) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(ci)</span>
<span id="cb29-146"><a href="#cb29-146" aria-hidden="true" tabindex="-1"></a>CIs</span>
<span id="cb29-147"><a href="#cb29-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-148"><a href="#cb29-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-149"><a href="#cb29-149" aria-hidden="true" tabindex="-1"></a>Nes <span class="ot">&lt;-</span> <span class="fu">rbind</span>(barplot.data[ , <span class="dv">1</span>],</span>
<span id="cb29-150"><a href="#cb29-150" aria-hidden="true" tabindex="-1"></a>             barplot.data[ , <span class="dv">2</span>],</span>
<span id="cb29-151"><a href="#cb29-151" aria-hidden="true" tabindex="-1"></a>             barplot.data[ , <span class="dv">3</span>])</span>
<span id="cb29-152"><a href="#cb29-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-153"><a href="#cb29-153" aria-hidden="true" tabindex="-1"></a>Nes</span>
<span id="cb29-154"><a href="#cb29-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-155"><a href="#cb29-155" aria-hidden="true" tabindex="-1"></a>bp</span>
<span id="cb29-156"><a href="#cb29-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-157"><a href="#cb29-157" aria-hidden="true" tabindex="-1"></a>idx <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb29-158"><a href="#cb29-158" aria-hidden="true" tabindex="-1"></a>idx2 <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb29-159"><a href="#cb29-159" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(Nes)){</span>
<span id="cb29-160"><a href="#cb29-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-161"><a href="#cb29-161" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrows</span>(<span class="at">x0 =</span> bp[i, ], <span class="at">y0 =</span> <span class="fu">abs</span>(Nes[i, ]<span class="sc">-</span>CIs[idx, ]),</span>
<span id="cb29-162"><a href="#cb29-162" aria-hidden="true" tabindex="-1"></a>           <span class="at">x1 =</span> bp[i, ], <span class="at">y1 =</span> <span class="fu">abs</span>(Nes[i, ]<span class="sc">+</span>CIs[idx2, ]),</span>
<span id="cb29-163"><a href="#cb29-163" aria-hidden="true" tabindex="-1"></a>           <span class="at">code =</span> <span class="dv">3</span>, <span class="at">angle =</span> <span class="dv">90</span>, <span class="at">length =</span> <span class="fl">0.04</span>, <span class="at">col =</span> <span class="st">"black"</span>)</span>
<span id="cb29-164"><a href="#cb29-164" aria-hidden="true" tabindex="-1"></a>    idx <span class="ot">=</span> idx <span class="sc">+</span> <span class="dv">2</span></span>
<span id="cb29-165"><a href="#cb29-165" aria-hidden="true" tabindex="-1"></a>    idx2 <span class="ot">=</span> idx2 <span class="sc">+</span> <span class="dv">2</span></span>
<span id="cb29-166"><a href="#cb29-166" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb29-167"><a href="#cb29-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-168"><a href="#cb29-168" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">"topright"</span>, <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">"0.05"</span>, <span class="st">"0.02"</span>, <span class="st">"0.01"</span>), <span class="at">density =</span> <span class="fu">c</span>(<span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">40</span>),</span>
<span id="cb29-169"><a href="#cb29-169" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"Critical value"</span>, <span class="at">bty =</span> <span class="st">"n"</span>, <span class="at">cex =</span> <span class="dv">1</span>, <span class="at">border =</span> <span class="cn">TRUE</span>)</span>
<span id="cb29-170"><a href="#cb29-170" aria-hidden="true" tabindex="-1"></a><span class="fu">mtext</span>(<span class="st">"A"</span>, <span class="at">side =</span> <span class="dv">2</span>, <span class="at">line =</span> <span class="fl">3.2</span>, <span class="at">at =</span> <span class="dv">150</span>, <span class="at">las =</span> <span class="dv">1</span>)</span>
<span id="cb29-171"><a href="#cb29-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-172"><a href="#cb29-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-173"><a href="#cb29-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-174"><a href="#cb29-174" aria-hidden="true" tabindex="-1"></a><span class="do">## plotting a map with GO values</span></span>
<span id="cb29-175"><a href="#cb29-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-176"><a href="#cb29-176" aria-hidden="true" tabindex="-1"></a><span class="do">## my colors - they might change the story!</span></span>
<span id="cb29-177"><a href="#cb29-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-178"><a href="#cb29-178" aria-hidden="true" tabindex="-1"></a>my.colors <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">"lightblue3"</span>, <span class="st">"orange2"</span>, <span class="st">"red3"</span>))(<span class="dv">100</span>)</span>
<span id="cb29-179"><a href="#cb29-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-180"><a href="#cb29-180" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="fl">3.1</span>, <span class="fl">4.1</span>, <span class="dv">0</span>, <span class="fl">2.1</span>), <span class="at">mgp =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="fl">0.8</span>, <span class="dv">0</span>))</span>
<span id="cb29-181"><a href="#cb29-181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-182"><a href="#cb29-182" aria-hidden="true" tabindex="-1"></a><span class="fu">image</span>(long.mat, lat.mat, go,</span>
<span id="cb29-183"><a href="#cb29-183" aria-hidden="true" tabindex="-1"></a>      <span class="at">col =</span> <span class="st">"white"</span>,</span>
<span id="cb29-184"><a href="#cb29-184" aria-hidden="true" tabindex="-1"></a>      <span class="at">las =</span> <span class="dv">1</span>,</span>
<span id="cb29-185"><a href="#cb29-185" aria-hidden="true" tabindex="-1"></a>      <span class="at">xlab =</span> <span class="st">"Longitude"</span>,</span>
<span id="cb29-186"><a href="#cb29-186" aria-hidden="true" tabindex="-1"></a>      <span class="at">ylab =</span> <span class="st">"Latitude"</span>,</span>
<span id="cb29-187"><a href="#cb29-187" aria-hidden="true" tabindex="-1"></a>      <span class="at">xlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">180</span>, <span class="sc">-</span><span class="dv">50</span>),</span>
<span id="cb29-188"><a href="#cb29-188" aria-hidden="true" tabindex="-1"></a>      <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">25</span>, <span class="dv">70</span>),</span>
<span id="cb29-189"><a href="#cb29-189" aria-hidden="true" tabindex="-1"></a>      <span class="at">smallplot =</span> <span class="cn">NULL</span>,</span>
<span id="cb29-190"><a href="#cb29-190" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend =</span> <span class="cn">NULL</span>,</span>
<span id="cb29-191"><a href="#cb29-191" aria-hidden="true" tabindex="-1"></a>      <span class="at">cex.axis =</span> <span class="dv">1</span>,</span>
<span id="cb29-192"><a href="#cb29-192" aria-hidden="true" tabindex="-1"></a>      <span class="at">cex.lab =</span> <span class="fl">1.5</span>)</span>
<span id="cb29-193"><a href="#cb29-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-194"><a href="#cb29-194" aria-hidden="true" tabindex="-1"></a><span class="fu">grid</span>()</span>
<span id="cb29-195"><a href="#cb29-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-196"><a href="#cb29-196" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">new =</span> <span class="cn">TRUE</span>)</span>
<span id="cb29-197"><a href="#cb29-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-198"><a href="#cb29-198" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="fl">3.1</span>, <span class="fl">4.1</span>, <span class="dv">0</span>, <span class="fl">2.1</span>), <span class="at">mgp =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="fl">0.8</span>, <span class="dv">0</span>))</span>
<span id="cb29-199"><a href="#cb29-199" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-200"><a href="#cb29-200" aria-hidden="true" tabindex="-1"></a><span class="fu">image</span>(long.mat, lat.mat, go,</span>
<span id="cb29-201"><a href="#cb29-201" aria-hidden="true" tabindex="-1"></a>      <span class="at">col =</span> my.colors,</span>
<span id="cb29-202"><a href="#cb29-202" aria-hidden="true" tabindex="-1"></a>      <span class="at">las =</span> <span class="dv">1</span>,</span>
<span id="cb29-203"><a href="#cb29-203" aria-hidden="true" tabindex="-1"></a>      <span class="at">xlab =</span> <span class="st">"Longitude"</span>,</span>
<span id="cb29-204"><a href="#cb29-204" aria-hidden="true" tabindex="-1"></a>      <span class="at">ylab =</span> <span class="st">"Latitude"</span>,</span>
<span id="cb29-205"><a href="#cb29-205" aria-hidden="true" tabindex="-1"></a>      <span class="at">xlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">180</span>, <span class="sc">-</span><span class="dv">50</span>),</span>
<span id="cb29-206"><a href="#cb29-206" aria-hidden="true" tabindex="-1"></a>      <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">25</span>, <span class="dv">70</span>),</span>
<span id="cb29-207"><a href="#cb29-207" aria-hidden="true" tabindex="-1"></a>      <span class="at">smallplot =</span> <span class="cn">NULL</span>,</span>
<span id="cb29-208"><a href="#cb29-208" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend =</span> <span class="cn">NULL</span>,</span>
<span id="cb29-209"><a href="#cb29-209" aria-hidden="true" tabindex="-1"></a>      <span class="at">cex.axis =</span> <span class="dv">1</span>,</span>
<span id="cb29-210"><a href="#cb29-210" aria-hidden="true" tabindex="-1"></a>      <span class="at">cex.lab =</span> <span class="fl">1.5</span>)</span>
<span id="cb29-211"><a href="#cb29-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-212"><a href="#cb29-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-213"><a href="#cb29-213" aria-hidden="true" tabindex="-1"></a>maps<span class="sc">::</span><span class="fu">map</span>(<span class="at">add =</span> <span class="cn">TRUE</span>, <span class="at">interior =</span> <span class="cn">FALSE</span>, <span class="at">col =</span> <span class="st">"grey40"</span>)</span>
<span id="cb29-214"><a href="#cb29-214" aria-hidden="true" tabindex="-1"></a><span class="fu">map</span>(<span class="st">"state"</span>, <span class="at">fill =</span> <span class="cn">TRUE</span>, <span class="at">col =</span> <span class="st">"transparent"</span>, <span class="at">bg =</span> <span class="st">"white"</span>, <span class="at">add =</span> <span class="cn">TRUE</span>)</span>
<span id="cb29-215"><a href="#cb29-215" aria-hidden="true" tabindex="-1"></a><span class="do">## getting Canada data (sf format)</span></span>
<span id="cb29-216"><a href="#cb29-216" aria-hidden="true" tabindex="-1"></a>canada_sf <span class="ot">&lt;-</span> <span class="fu">ne_states</span>(<span class="at">country =</span> <span class="st">"Canada"</span>, <span class="at">returnclass =</span> <span class="st">"sf"</span>)</span>
<span id="cb29-217"><a href="#cb29-217" aria-hidden="true" tabindex="-1"></a><span class="fu">map</span>(canada_sf, <span class="at">fill =</span> <span class="cn">TRUE</span>, <span class="at">col =</span> <span class="st">"transparent"</span>, <span class="at">bg =</span> <span class="st">"white"</span>, <span class="at">add =</span> <span class="cn">TRUE</span>)</span>
<span id="cb29-218"><a href="#cb29-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-219"><a href="#cb29-219" aria-hidden="true" tabindex="-1"></a><span class="do">## plotting samples.  Dot sizes correspond to values of Nes</span></span>
<span id="cb29-220"><a href="#cb29-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-221"><a href="#cb29-221" aria-hidden="true" tabindex="-1"></a><span class="fu">with</span>(data, <span class="fu">points</span>(longitude, latitude, <span class="at">pch =</span> <span class="dv">21</span>,</span>
<span id="cb29-222"><a href="#cb29-222" aria-hidden="true" tabindex="-1"></a>                  <span class="at">bg =</span> RColorBrewer<span class="sc">::</span><span class="fu">brewer.pal</span>(<span class="dv">9</span>,<span class="st">"Paired"</span>)[Pops],</span>
<span id="cb29-223"><a href="#cb29-223" aria-hidden="true" tabindex="-1"></a>                  <span class="at">col =</span> <span class="fu">alpha</span>(<span class="st">"black"</span>, <span class="fl">0.3</span>)))</span>
<span id="cb29-224"><a href="#cb29-224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-225"><a href="#cb29-225" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="at">x =</span> <span class="sc">-</span><span class="dv">180</span>, <span class="at">y =</span> <span class="dv">50</span>, <span class="at">legend =</span> <span class="fu">levels</span>(data<span class="sc">$</span>Pops), <span class="at">pch =</span> <span class="dv">19</span>,</span>
<span id="cb29-226"><a href="#cb29-226" aria-hidden="true" tabindex="-1"></a>       <span class="at">col =</span> RColorBrewer<span class="sc">::</span><span class="fu">brewer.pal</span>(<span class="dv">9</span>,<span class="st">"Paired"</span>), <span class="at">bty =</span> <span class="st">"n"</span>,</span>
<span id="cb29-227"><a href="#cb29-227" aria-hidden="true" tabindex="-1"></a>       <span class="at">cex =</span> <span class="dv">1</span>)</span>
<span id="cb29-228"><a href="#cb29-228" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-229"><a href="#cb29-229" aria-hidden="true" tabindex="-1"></a><span class="fu">addnortharrow</span>(<span class="at">pos =</span> <span class="st">"bottomleft"</span>, <span class="at">scale =</span> <span class="fl">0.4</span>)</span>
<span id="cb29-230"><a href="#cb29-230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-231"><a href="#cb29-231" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">new =</span> <span class="cn">TRUE</span>)</span>
<span id="cb29-232"><a href="#cb29-232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-233"><a href="#cb29-233" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="fl">3.1</span>, <span class="fl">4.1</span>, <span class="dv">0</span>, <span class="fl">2.1</span>), <span class="at">mgp =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="fl">0.8</span>, <span class="dv">0</span>))</span>
<span id="cb29-234"><a href="#cb29-234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-235"><a href="#cb29-235" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">scipen =</span> <span class="sc">-</span><span class="dv">100</span>)</span>
<span id="cb29-236"><a href="#cb29-236" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">digits =</span> <span class="dv">2</span>)</span>
<span id="cb29-237"><a href="#cb29-237" aria-hidden="true" tabindex="-1"></a>fields<span class="sc">::</span><span class="fu">image.plot</span>(long.mat, lat.mat, go,</span>
<span id="cb29-238"><a href="#cb29-238" aria-hidden="true" tabindex="-1"></a>                   <span class="at">legend.only =</span> <span class="cn">TRUE</span>,</span>
<span id="cb29-239"><a href="#cb29-239" aria-hidden="true" tabindex="-1"></a>                   <span class="at">horizontal =</span> <span class="cn">TRUE</span>,</span>
<span id="cb29-240"><a href="#cb29-240" aria-hidden="true" tabindex="-1"></a>                   <span class="at">legend.args =</span> <span class="fu">list</span>(<span class="at">text =</span> <span class="st">"Genomic offset"</span>, <span class="at">side =</span> <span class="dv">3</span>,</span>
<span id="cb29-241"><a href="#cb29-241" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">line =</span> <span class="fl">0.5</span>, <span class="at">cex =</span> <span class="fl">0.8</span>),</span>
<span id="cb29-242"><a href="#cb29-242" aria-hidden="true" tabindex="-1"></a>                   <span class="at">axis.args =</span> <span class="fu">list</span>(<span class="at">cex.axis =</span> <span class="fl">0.56</span>, <span class="at">mgp =</span> <span class="fu">c</span>(<span class="dv">3</span>, <span class="fl">0.5</span>, <span class="dv">0</span>)), </span>
<span id="cb29-243"><a href="#cb29-243" aria-hidden="true" tabindex="-1"></a>                   <span class="at">smallplot =</span> <span class="fu">c</span>(<span class="fl">0.19</span>, <span class="fl">0.43</span>, <span class="fl">0.28</span>, <span class="fl">0.29</span>), <span class="at">col =</span> my.colors)</span>
<span id="cb29-244"><a href="#cb29-244" aria-hidden="true" tabindex="-1"></a><span class="fu">mtext</span>(<span class="st">"B"</span>, <span class="at">side =</span> <span class="dv">2</span>, <span class="at">line =</span> <span class="dv">3</span>, <span class="at">at =</span> <span class="dv">79</span>, <span class="at">las =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><img src="imgs/Ne_GO.png" class="img-fluid"></p>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>